// @web3modal/ui-react-native/lib/commonjs/layout/wui-flex/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.FlexView = FlexView;
var _reactNative = require("react-native");
var _UiUtil = require("../../utils/UiUtil");
function FlexView(props) {
  const styles = {
    flexDirection: props.flexDirection,
    flexWrap: props.flexWrap,
    flexGrow: props.flexGrow,
    flexShrink: props.flexShrink,
    alignItems: props.alignItems,
    alignSelf: props.alignSelf,
    justifyContent: props.justifyContent,
    paddingTop: props.padding && _UiUtil.UiUtil.getSpacingStyles(props.padding, 0),
    paddingRight: props.padding && _UiUtil.UiUtil.getSpacingStyles(props.padding, 1),
    paddingBottom: props.padding && _UiUtil.UiUtil.getSpacingStyles(props.padding, 2),
    paddingLeft: props.padding && _UiUtil.UiUtil.getSpacingStyles(props.padding, 3),
    marginTop: props.margin && _UiUtil.UiUtil.getSpacingStyles(props.margin, 0),
    marginRight: props.margin && _UiUtil.UiUtil.getSpacingStyles(props.margin, 1),
    marginBottom: props.margin && _UiUtil.UiUtil.getSpacingStyles(props.margin, 2),
    marginLeft: props.margin && _UiUtil.UiUtil.getSpacingStyles(props.margin, 3)
  };
  return /*#__PURE__*/React.createElement(_reactNative.View, {
    style: [styles, props.style]
  }, props.children);
}
