// @web3modal/ui-react-native/lib/commonjs/components/wui-shimmer/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Shimmer = void 0;
var _reactNativeSvg = require("react-native-svg");
var _reactNative = require("react-native");
var _useTheme = require("../../hooks/useTheme");
const AnimatedRect = _reactNative.Animated.createAnimatedComponent(_reactNativeSvg.Rect);
const Shimmer = ({
  width = 200,
  height = 200,
  duration = 1000,
  borderRadius = 0,
  backgroundColor,
  foregroundColor,
  style
}) => {
  const animatedValue = new _reactNative.Animated.Value(0);
  const Theme = (0, _useTheme.useTheme)();
  const animatedProps = {
    fill: animatedValue.interpolate({
      inputRange: [0, 0.5, 1],
      outputRange: [backgroundColor ?? Theme['bg-200'], foregroundColor ?? Theme['bg-300'], backgroundColor ?? Theme['bg-200']]
    }),
    width,
    height,
    x: 0,
    y: 0,
    rx: borderRadius,
    ry: borderRadius
  };
  _reactNative.Animated.loop(_reactNative.Animated.timing(animatedValue, {
    toValue: 1,
    duration,
    useNativeDriver: false
  })).start();
  return /*#__PURE__*/React.createElement(_reactNativeSvg.Svg, {
    width: width,
    height: height,
    style: style
  }, /*#__PURE__*/React.createElement(AnimatedRect, animatedProps));
};
exports.Shimmer = Shimmer;
