// @web3modal/ui-react-native/lib/module/composites/wui-avatar/index.js
import { View } from 'react-native';
import Svg, { Circle, Defs, RadialGradient, Stop } from 'react-native-svg';
import { Image } from '../../components/wui-image';
import { useTheme } from '../../hooks/useTheme';
import { UiUtil } from '../../utils/UiUtil';
import styles from './styles';
export function Avatar({
  imageSrc,
  address,
  style,
  size = 64,
  borderWidth = 8
}) {
  const Theme = useTheme();
  const colors = UiUtil.generateAvatarColors(address);
  const containerSize = size + borderWidth * 2;
  if (imageSrc) {
    return /*#__PURE__*/React.createElement(View, {
      style: [styles.image, {
        height: containerSize,
        width: containerSize,
        borderColor: Theme['gray-glass-005'],
        borderWidth
      }, style]
    }, /*#__PURE__*/React.createElement(Image, {
      source: imageSrc,
      style: [styles.image, {
        height: size,
        width: size
      }]
    }));
  }
  return /*#__PURE__*/React.createElement(Svg, {
    width: containerSize,
    height: containerSize,
    fill: "none",
    style: style
  }, /*#__PURE__*/React.createElement(Defs, null, /*#__PURE__*/React.createElement(RadialGradient, {
    id: "radial",
    cx: "0",
    cy: "0",
    r: "1",
    gradientUnits: "userSpaceOnUse",
    gradientTransform: `translate(${size * 0.75} ${size * 0.35}) scale(${size * 0.75})`
  }, /*#__PURE__*/React.createElement(Stop, {
    offset: "0.52%",
    stopColor: "#FFFFFF"
  }), /*#__PURE__*/React.createElement(Stop, {
    offset: "31.25%",
    stopColor: colors[0]
  }), /*#__PURE__*/React.createElement(Stop, {
    offset: "51.56%",
    stopColor: colors[1]
  }), /*#__PURE__*/React.createElement(Stop, {
    offset: "65.63%",
    stopColor: colors[2]
  }), /*#__PURE__*/React.createElement(Stop, {
    offset: "82.29%",
    stopColor: colors[3]
  }), /*#__PURE__*/React.createElement(Stop, {
    offset: "100%",
    stopColor: colors[4]
  }))), /*#__PURE__*/React.createElement(Circle, {
    cx: "50%",
    cy: "50%",
    r: size / 2,
    fill: "url(#radial)"
  }), /*#__PURE__*/React.createElement(Circle, {
    cx: "50%",
    cy: "50%",
    r: (size + borderWidth) / 2,
    stroke: Theme['gray-glass-005'],
    strokeWidth: borderWidth
  }));
}
