// @web3modal/scaffold-react-native/lib/module/modal/w3m-account-button/index.js
import { useSnapshot } from 'valtio';
import { AccountController, CoreHelperUtil, NetworkController, ModalController, AssetUtil } from '@web3modal/core-react-native';
import { AccountButton } from '@web3modal/ui-react-native';
import { ApiController } from '@web3modal/core-react-native';
export function W3mAccountButton({
  balance,
  disabled,
  style,
  testID
}) {
  const {
    address,
    balance: balanceVal,
    balanceSymbol,
    profileImage,
    profileName
  } = useSnapshot(AccountController.state);
  const {
    caipNetwork
  } = useSnapshot(NetworkController.state);
  const networkImage = AssetUtil.getNetworkImage(caipNetwork);
  const showBalance = balance === 'show';
  return /*#__PURE__*/React.createElement(AccountButton, {
    onPress: ModalController.open,
    address: profileName ?? address,
    isProfileName: Boolean(profileName),
    networkSrc: networkImage,
    imageHeaders: ApiController._getApiHeaders(),
    avatarSrc: profileImage,
    disabled: disabled,
    style: style,
    balance: showBalance ? CoreHelperUtil.formatBalance(balanceVal, balanceSymbol) : '',
    testID: testID
  });
}
