// @web3modal/ui-react-native/lib/module/composites/wui-list-item/index.js
import { View, Pressable, Animated } from 'react-native';
import { Icon } from '../../components/wui-icon';
import { Image } from '../../components/wui-image';
import { LoadingSpinner } from '../../components/wui-loading-spinner';
import useAnimatedValue from '../../hooks/useAnimatedValue';
import { useTheme } from '../../hooks/useTheme';
import { IconBox } from '../wui-icon-box';
import styles from './styles';
const AnimatedPressable = Animated.createAnimatedComponent(Pressable);
export function ListItem({
  children,
  icon,
  variant,
  imageSrc,
  imageHeaders,
  iconVariant = 'blue',
  chevron,
  loading,
  disabled,
  onPress,
  style,
  testID
}) {
  const Theme = useTheme();
  const {
    animatedValue,
    setStartValue,
    setEndValue
  } = useAnimatedValue(Theme['gray-glass-002'], Theme['gray-glass-010']);
  function visualTemplate() {
    if (variant === 'image' && imageSrc) {
      return /*#__PURE__*/React.createElement(View, {
        style: [styles.imageContainer, {
          borderColor: Theme['gray-glass-005']
        }]
      }, /*#__PURE__*/React.createElement(Image, {
        source: imageSrc,
        headers: imageHeaders,
        style: [styles.image, (disabled || loading) && styles.disabledImage]
      }));
    } else if (variant === 'icon' && icon) {
      const iconColor = iconVariant === 'blue' ? 'accent-100' : 'fg-200';
      const borderColor = iconVariant === 'blue' ? 'accent-glass-005' : 'gray-glass-005';
      return /*#__PURE__*/React.createElement(View, {
        style: [styles.imageContainer, {
          borderColor: Theme[borderColor]
        }]
      }, /*#__PURE__*/React.createElement(IconBox, {
        icon: icon,
        iconColor: iconColor,
        size: "md",
        background: true,
        backgroundColor: "gray-glass-010"
      }));
    }
    return null;
  }
  function rightTemplate() {
    if (loading) {
      return /*#__PURE__*/React.createElement(LoadingSpinner, {
        color: "fg-200",
        size: "lg",
        style: styles.rightIcon
      });
    } else if (chevron) {
      return /*#__PURE__*/React.createElement(Icon, {
        name: "chevronRight",
        size: "sm",
        color: "fg-200",
        style: styles.rightIcon
      });
    }
    return null;
  }
  return /*#__PURE__*/React.createElement(AnimatedPressable, {
    disabled: disabled || loading,
    style: [styles.container, {
      backgroundColor: disabled || loading ? Theme['gray-glass-010'] : animatedValue
    }, style],
    onPress: onPress,
    onPressIn: setEndValue,
    onPressOut: setStartValue,
    testID: testID
  }, visualTemplate(), /*#__PURE__*/React.createElement(View, {
    style: styles.content
  }, children), rightTemplate());
}
