// @web3modal/ui-react-native/lib/commonjs/composites/wui-network-image/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.NetworkImage = NetworkImage;
var _reactNativeSvg = require("react-native-svg");
var _useTheme = require("../../hooks/useTheme");
var _styles = require("./styles");
function NetworkImage({
  imageSrc,
  imageHeaders,
  disabled,
  selected,
  size = 'md'
}) {
  const Theme = (0, _useTheme.useTheme)();
  const isLg = size === 'lg';
  const svgWidth = isLg ? 96 : 56;
  const svgHeight = isLg ? 96 : 56;
  const svgStroke = selected ? Theme['accent-100'] : Theme['gray-glass-010'];
  const opacity = disabled ? 0.5 : 1;
  return /*#__PURE__*/React.createElement(_reactNativeSvg.Svg, {
    width: svgWidth,
    height: svgHeight,
    stroke: svgStroke,
    strokeWidth: 1
  }, /*#__PURE__*/React.createElement(_reactNativeSvg.Defs, null, /*#__PURE__*/React.createElement(_reactNativeSvg.Pattern, {
    id: "image-pattern",
    x: "0",
    y: "0",
    width: "1",
    height: "1"
  }, /*#__PURE__*/React.createElement(_reactNativeSvg.Image, {
    height: svgHeight,
    width: svgWidth,
    x: "0",
    y: "0",
    opacity: opacity,
    href: {
      uri: imageSrc,
      headers: imageHeaders
    }
  }))), /*#__PURE__*/React.createElement(_reactNativeSvg.Path, {
    d: `${isLg ? _styles.PathLg : _styles.PathNormal}`,
    opacity: opacity,
    fill: imageSrc ? 'url(#image-pattern)' : Theme['gray-glass-005']
  }));
}
