// @web3modal/scaffold-react-native/lib/commonjs/partials/w3m-connecting-mobile/styles.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _uiReactNative = require("@web3modal/ui-react-native");
var _reactNative = require("react-native");
var _default = exports.default = _reactNative.StyleSheet.create({
  container: {
    paddingBottom: _uiReactNative.Spacing['3xl']
  },
  retryButton: {
    marginTop: _uiReactNative.Spacing.m
  },
  retryIcon: {
    transform: [{
      rotateY: '180deg'
    }]
  },
  copyButton: {
    alignSelf: 'center',
    marginTop: _uiReactNative.Spacing.m
  },
  textContainer: {
    marginVertical: _uiReactNative.Spacing.xs
  },
  descriptionText: {
    marginTop: _uiReactNative.Spacing.xs,
    marginHorizontal: _uiReactNative.Spacing['3xl']
  },
  errorIcon: {
    position: 'absolute',
    bottom: 5,
    right: 5,
    zIndex: 2
  },
  storeButton: {
    justifyContent: 'space-between',
    paddingHorizontal: _uiReactNative.Spacing.l,
    marginHorizontal: _uiReactNative.Spacing.xl,
    marginTop: _uiReactNative.Spacing.l
  }
});
