// @web3modal/ui-react-native/lib/commonjs/composites/wui-wallet-image/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.WalletImage = WalletImage;
var _reactNative = require("react-native");
var _wuiIcon = require("../../components/wui-icon");
var _wuiImage = require("../../components/wui-image");
var _useTheme = require("../../hooks/useTheme");
var _styles = _interopRequireDefault(require("./styles"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
const IconSize = {
  xs: 10,
  sm: 24,
  md: 32,
  lg: 40
};
function WalletImage({
  imageSrc,
  imageHeaders,
  size = 'md',
  style,
  showAllWallets
}) {
  const Theme = (0, _useTheme.useTheme)();
  const templateVisual = () => {
    if (imageSrc) {
      return /*#__PURE__*/React.createElement(_wuiImage.Image, {
        source: imageSrc,
        headers: imageHeaders,
        style: _styles.default[`${size}Image`]
      });
    }
    if (showAllWallets) {
      return /*#__PURE__*/React.createElement(_reactNative.View, {
        style: [_styles.default.container, _styles.default[`${size}Image`]]
      }, /*#__PURE__*/React.createElement(_wuiIcon.Icon, {
        height: IconSize[size],
        width: IconSize[size],
        name: "allWallets",
        color: "accent-100"
      }));
    }
    return /*#__PURE__*/React.createElement(_reactNative.View, {
      style: [_styles.default.container, _styles.default[`${size}Image`]]
    }, /*#__PURE__*/React.createElement(_wuiIcon.Icon, {
      height: IconSize[size],
      width: IconSize[size],
      name: "walletPlaceholder",
      color: "fg-200"
    }));
  };
  return /*#__PURE__*/React.createElement(_reactNative.View, {
    style: style
  }, templateVisual(), /*#__PURE__*/React.createElement(_reactNative.View, {
    style: [_styles.default.border, _styles.default[`${size}Image`], {
      borderColor: Theme['gray-glass-010']
    }, !imageSrc && {
      backgroundColor: Theme['gray-glass-002']
    }, showAllWallets && {
      backgroundColor: `${Theme['accent-100']}1E`,
      borderColor: Theme['accent-glass-010']
    }]
  }));
}
