// @web3modal/ui-react-native/lib/commonjs/composites/wui-wallet-image/styles.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _reactNative = require("react-native");
var _ThemeUtil = require("../../utils/ThemeUtil");
var _default = exports.default = _reactNative.StyleSheet.create({
  container: {
    alignItems: 'center',
    justifyContent: 'center'
  },
  xsImage: {
    height: _ThemeUtil.WalletImageSize.xs,
    width: _ThemeUtil.WalletImageSize.xs,
    borderRadius: _ThemeUtil.BorderRadius['4xs']
  },
  smImage: {
    height: _ThemeUtil.WalletImageSize.sm,
    width: _ThemeUtil.WalletImageSize.sm,
    borderRadius: _ThemeUtil.BorderRadius.xxs
  },
  mdImage: {
    height: _ThemeUtil.WalletImageSize.md,
    width: _ThemeUtil.WalletImageSize.md,
    borderRadius: _ThemeUtil.BorderRadius.xs
  },
  lgImage: {
    height: _ThemeUtil.WalletImageSize.lg,
    width: _ThemeUtil.WalletImageSize.lg,
    borderRadius: _ThemeUtil.BorderRadius.m
  },
  border: {
    borderWidth: 1,
    position: 'absolute'
  }
});
