// @web3modal/ui-react-native/lib/commonjs/composites/wui-list-item/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ListItem = ListItem;
var _reactNative = require("react-native");
var _wuiIcon = require("../../components/wui-icon");
var _wuiImage = require("../../components/wui-image");
var _wuiLoadingSpinner = require("../../components/wui-loading-spinner");
var _useAnimatedValue = _interopRequireDefault(require("../../hooks/useAnimatedValue"));
var _useTheme = require("../../hooks/useTheme");
var _wuiIconBox = require("../wui-icon-box");
var _styles = _interopRequireDefault(require("./styles"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
const AnimatedPressable = _reactNative.Animated.createAnimatedComponent(_reactNative.Pressable);
function ListItem({
  children,
  icon,
  variant,
  imageSrc,
  imageHeaders,
  iconVariant = 'blue',
  chevron,
  loading,
  disabled,
  onPress,
  style,
  testID
}) {
  const Theme = (0, _useTheme.useTheme)();
  const {
    animatedValue,
    setStartValue,
    setEndValue
  } = (0, _useAnimatedValue.default)(Theme['gray-glass-002'], Theme['gray-glass-010']);
  function visualTemplate() {
    if (variant === 'image' && imageSrc) {
      return /*#__PURE__*/React.createElement(_reactNative.View, {
        style: [_styles.default.imageContainer, {
          borderColor: Theme['gray-glass-005']
        }]
      }, /*#__PURE__*/React.createElement(_wuiImage.Image, {
        source: imageSrc,
        headers: imageHeaders,
        style: [_styles.default.image, (disabled || loading) && _styles.default.disabledImage]
      }));
    } else if (variant === 'icon' && icon) {
      const iconColor = iconVariant === 'blue' ? 'accent-100' : 'fg-200';
      const borderColor = iconVariant === 'blue' ? 'accent-glass-005' : 'gray-glass-005';
      return /*#__PURE__*/React.createElement(_reactNative.View, {
        style: [_styles.default.imageContainer, {
          borderColor: Theme[borderColor]
        }]
      }, /*#__PURE__*/React.createElement(_wuiIconBox.IconBox, {
        icon: icon,
        iconColor: iconColor,
        size: "md",
        background: true,
        backgroundColor: "gray-glass-010"
      }));
    }
    return null;
  }
  function rightTemplate() {
    if (loading) {
      return /*#__PURE__*/React.createElement(_wuiLoadingSpinner.LoadingSpinner, {
        color: "fg-200",
        size: "lg",
        style: _styles.default.rightIcon
      });
    } else if (chevron) {
      return /*#__PURE__*/React.createElement(_wuiIcon.Icon, {
        name: "chevronRight",
        size: "sm",
        color: "fg-200",
        style: _styles.default.rightIcon
      });
    }
    return null;
  }
  return /*#__PURE__*/React.createElement(AnimatedPressable, {
    disabled: disabled || loading,
    style: [_styles.default.container, {
      backgroundColor: disabled || loading ? Theme['gray-glass-010'] : animatedValue
    }, style],
    onPress: onPress,
    onPressIn: setEndValue,
    onPressOut: setStartValue,
    testID: testID
  }, visualTemplate(), /*#__PURE__*/React.createElement(_reactNative.View, {
    style: _styles.default.content
  }, children), rightTemplate());
}
