// @web3modal/scaffold-react-native/lib/commonjs/partials/w3m-connecting-header/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ConnectingHeader = ConnectingHeader;
var _uiReactNative = require("@web3modal/ui-react-native");
function ConnectingHeader({
  platforms,
  onSelectPlatform
}) {
  const generateTabs = () => {
    const tabs = platforms.map(platform => {
      if (platform === 'mobile') {
        return {
          label: 'Mobile',
          icon: 'mobile',
          platform: 'mobile'
        };
      } else if (platform === 'web') {
        return {
          label: 'Web',
          icon: 'browser',
          platform: 'web'
        };
      } else {
        return {
          label: 'QR Code',
          icon: 'qrCode',
          platform: 'qrcode'
        };
      }
    });
    return tabs;
  };
  const onTabChange = index => {
    const platform = platforms[index];
    if (platform) {
      onSelectPlatform(platform);
    }
  };
  const tabs = generateTabs();
  return /*#__PURE__*/React.createElement(_uiReactNative.FlexView, {
    alignItems: "center",
    padding: ['xs', '0', '0', '0']
  }, /*#__PURE__*/React.createElement(_uiReactNative.Tabs, {
    tabs: tabs,
    onTabChange: onTabChange
  }));
}
