// @web3modal/scaffold-react-native/lib/commonjs/views/w3m-get-wallet-view/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.GetWalletView = GetWalletView;
var _valtio = require("valtio");
var _reactNative = require("react-native");
var _uiReactNative = require("@web3modal/ui-react-native");
var _coreReactNative = require("@web3modal/core-react-native");
var _useCustomDimensions = require("../../hooks/useCustomDimensions");
var _styles = _interopRequireDefault(require("./styles"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function GetWalletView() {
  const {
    padding
  } = (0, _useCustomDimensions.useCustomDimensions)();
  const {
    recommended
  } = (0, _valtio.useSnapshot)(_coreReactNative.ApiController.state);
  const imageHeaders = _coreReactNative.ApiController._getApiHeaders();
  const onWalletPress = wallet => {
    const storeUrl = _reactNative.Platform.select({
      ios: wallet.app_store,
      android: wallet.play_store
    }) || wallet.homepage;
    if (storeUrl) {
      _reactNative.Linking.openURL(storeUrl);
    }
  };
  const listTemplate = () => {
    return recommended.map(wallet => /*#__PURE__*/React.createElement(_uiReactNative.ListWallet, {
      key: wallet.id,
      name: wallet.name,
      imageSrc: _coreReactNative.AssetUtil.getWalletImage(wallet),
      imageHeaders: imageHeaders,
      onPress: () => onWalletPress(wallet),
      style: _styles.default.item
    }));
  };
  return /*#__PURE__*/React.createElement(_reactNative.ScrollView, {
    bounces: false,
    style: {
      paddingHorizontal: padding
    },
    fadingEdgeLength: 20
  }, /*#__PURE__*/React.createElement(_uiReactNative.FlexView, {
    padding: ['s', 's', '3xl', 's']
  }, listTemplate(), /*#__PURE__*/React.createElement(_uiReactNative.ListWallet, {
    name: "Explore all",
    showAllWallets: true,
    icon: "externalLink",
    onPress: () => _reactNative.Linking.openURL('https://walletconnect.com/explorer?type=wallet')
  })));
}
