// @web3modal/core-react-native/lib/commonjs/controllers/AccountController.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.AccountController = void 0;
var _valtio = require("valtio");
var _utils = require("valtio/utils");
var _CoreHelperUtil = require("../utils/CoreHelperUtil");
// -- Types --------------------------------------------- //

// -- State --------------------------------------------- //
const state = (0, _valtio.proxy)({
  isConnected: false
});

// -- Controller ---------------------------------------- //
const AccountController = exports.AccountController = {
  state,
  subscribeConnection(callback) {
    return (0, _utils.subscribeKey)(state, 'isConnected', callback);
  },
  setIsConnected(isConnected) {
    state.isConnected = isConnected;
  },
  setCaipAddress(caipAddress) {
    const address = caipAddress ? _CoreHelperUtil.CoreHelperUtil.getPlainAddress(caipAddress) : undefined;
    state.caipAddress = caipAddress;
    state.address = address;
  },
  setBalance(balance, balanceSymbol) {
    state.balance = balance;
    state.balanceSymbol = balanceSymbol;
  },
  setProfileName(profileName) {
    state.profileName = profileName;
  },
  setProfileImage(profileImage) {
    state.profileImage = profileImage;
  },
  setAddressExplorerUrl(explorerUrl) {
    state.addressExplorerUrl = explorerUrl;
  },
  resetAccount() {
    state.isConnected = false;
    state.caipAddress = undefined;
    state.address = undefined;
    state.balance = undefined;
    state.balanceSymbol = undefined;
    state.profileName = undefined;
    state.profileImage = undefined;
    state.addressExplorerUrl = undefined;
  }
};
