// @web3modal/core-react-native/lib/commonjs/controllers/ModalController.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ModalController = void 0;
var _valtio = require("valtio");
var _AccountController = require("./AccountController");
var _RouterController = require("./RouterController");
var _PublicStateController = require("./PublicStateController");
var _EventsController = require("./EventsController");
var _ApiController = require("./ApiController");
// -- Types --------------------------------------------- //

// -- State --------------------------------------------- //
const state = (0, _valtio.proxy)({
  open: false,
  loading: false
});

// -- Controller ---------------------------------------- //
const ModalController = exports.ModalController = {
  state,
  async open(options) {
    await _ApiController.ApiController.state.prefetchPromise;
    const connected = _AccountController.AccountController.state.isConnected;
    if (options?.view) {
      _RouterController.RouterController.reset(options.view);
    } else if (_AccountController.AccountController.state.isConnected) {
      _RouterController.RouterController.reset('Account');
    } else {
      _RouterController.RouterController.reset('Connect');
    }
    state.open = true;
    _PublicStateController.PublicStateController.set({
      open: true
    });
    _EventsController.EventsController.sendEvent({
      type: 'track',
      event: 'MODAL_OPEN',
      properties: {
        connected
      }
    });
  },
  close() {
    const connected = _AccountController.AccountController.state.isConnected;
    state.open = false;
    _PublicStateController.PublicStateController.set({
      open: false
    });
    _EventsController.EventsController.sendEvent({
      type: 'track',
      event: 'MODAL_CLOSE',
      properties: {
        connected
      }
    });
  },
  setLoading(loading) {
    state.loading = loading;
  }
};
