// @web3modal/core-react-native/lib/commonjs/controllers/NetworkController.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.NetworkController = void 0;
var _valtio = require("valtio");
var _PublicStateController = require("./PublicStateController");
// -- Types --------------------------------------------- //

// -- State --------------------------------------------- //
const state = (0, _valtio.proxy)({
  supportsAllNetworks: true,
  isDefaultCaipNetwork: false
});

// -- Controller ---------------------------------------- //
const NetworkController = exports.NetworkController = {
  state,
  _getClient() {
    if (!state._client) {
      throw new Error('NetworkController client not set');
    }
    return state._client;
  },
  setClient(client) {
    state._client = (0, _valtio.ref)(client);
  },
  setCaipNetwork(caipNetwork) {
    state.caipNetwork = caipNetwork;
    _PublicStateController.PublicStateController.set({
      selectedNetworkId: caipNetwork?.id
    });
  },
  setDefaultCaipNetwork(caipNetwork) {
    state.caipNetwork = caipNetwork;
    state.isDefaultCaipNetwork = true;
    _PublicStateController.PublicStateController.set({
      selectedNetworkId: caipNetwork?.id
    });
  },
  setRequestedCaipNetworks(requestedNetworks) {
    state.requestedCaipNetworks = requestedNetworks;
  },
  async getApprovedCaipNetworksData() {
    const data = await this._getClient().getApprovedCaipNetworksData();
    state.supportsAllNetworks = data.supportsAllNetworks;
    state.approvedCaipNetworkIds = data.approvedCaipNetworkIds;
  },
  async switchActiveNetwork(network) {
    await this._getClient().switchCaipNetwork(network);
    state.caipNetwork = network;
    _PublicStateController.PublicStateController.set({
      selectedNetworkId: network?.id
    });
  },
  resetNetwork() {
    if (!state.isDefaultCaipNetwork) {
      state.caipNetwork = undefined;
    }
    state.approvedCaipNetworkIds = undefined;
    state.supportsAllNetworks = true;
  }
};
