// @web3modal/core-react-native/lib/commonjs/controllers/RouterController.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.RouterController = void 0;
var _valtio = require("valtio");
// -- Types --------------------------------------------- //

// -- State --------------------------------------------- //
const state = (0, _valtio.proxy)({
  view: 'Connect',
  history: ['Connect']
});

// -- Controller ---------------------------------------- //
const RouterController = exports.RouterController = {
  state,
  push(view, data) {
    if (view !== state.view) {
      state.view = view;
      state.history.push(view);
      state.data = data;
    }
  },
  reset(view) {
    state.view = view;
    state.history = [view];
  },
  replace(view) {
    if (state.history.length > 1 && state.history.at(-1) !== view) {
      state.view = view;
      state.history[state.history.length - 1] = view;
    }
  },
  goBack() {
    if (state.history.length > 1) {
      state.history.pop();
      const [last] = state.history.slice(-1);
      if (last) {
        state.view = last;
      }
    }
  }
};
