// @web3modal/ui-react-native/lib/module/composites/wui-search-bar/index.js
import { useRef, useState } from 'react';
import { InputElement } from '../wui-input-element';
import { InputText } from '../wui-input-text';
import { Spacing } from '../../utils/ThemeUtil';
export function SearchBar({
  placeholder = 'Search wallet',
  onSubmitEditing,
  onChangeText,
  inputStyle,
  testID
}) {
  const [showClear, setShowClear] = useState(false);
  const inputRef = useRef(null);
  const handleChangeText = text => {
    setShowClear(!!text?.length);
    onChangeText?.(text);
  };
  return /*#__PURE__*/React.createElement(InputText, {
    ref: inputRef,
    placeholder: placeholder,
    onChangeText: handleChangeText,
    onSubmitEditing: onSubmitEditing,
    icon: "search",
    inputStyle: inputStyle,
    returnKeyType: "search",
    disableFullscreenUI: true,
    testID: testID
  }, showClear && /*#__PURE__*/React.createElement(InputElement, {
    icon: "close",
    style: {
      marginRight: Spacing['4xs']
    },
    onPress: () => {
      inputRef.current?.clear();
      inputRef.current?.focus();
      handleChangeText('');
    }
  }));
}
