// @web3modal/ui-react-native/lib/commonjs/assets/visual/Network.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _reactNativeSvg = _interopRequireWildcard(require("react-native-svg"));
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && Object.prototype.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
const SvgNetwork = props => /*#__PURE__*/React.createElement(_reactNativeSvg.default, _extends({
  fill: "none",
  viewBox: "0 0 60 60"
}, props), /*#__PURE__*/React.createElement(_reactNativeSvg.G, {
  clipPath: "url(#a)"
}, /*#__PURE__*/React.createElement(_reactNativeSvg.Rect, {
  width: "60",
  height: "60",
  fill: "#00ACE6",
  rx: "30"
}), /*#__PURE__*/React.createElement(_reactNativeSvg.Circle, {
  cx: "64",
  cy: "39",
  r: "50",
  fill: "#1AC6FF",
  stroke: "#fff",
  strokeWidth: "2"
}), /*#__PURE__*/React.createElement(_reactNativeSvg.Circle, {
  cx: "78",
  cy: "30",
  r: "50",
  fill: "#4DD2FF",
  stroke: "#fff",
  strokeWidth: "2"
}), /*#__PURE__*/React.createElement(_reactNativeSvg.Circle, {
  cx: "72",
  cy: "15",
  r: "35",
  fill: "#80DFFF",
  stroke: "#fff",
  strokeWidth: "2"
}), /*#__PURE__*/React.createElement(_reactNativeSvg.Circle, {
  cx: "34",
  cy: "-17",
  r: "45",
  stroke: "#fff",
  strokeWidth: "2"
}), /*#__PURE__*/React.createElement(_reactNativeSvg.Circle, {
  cx: "34",
  cy: "-5",
  r: "50",
  stroke: "#fff",
  strokeWidth: "2"
}), /*#__PURE__*/React.createElement(_reactNativeSvg.Circle, {
  cx: "30",
  cy: "45",
  r: "4",
  fill: "#4DD2FF",
  stroke: "#fff",
  strokeWidth: "2"
}), /*#__PURE__*/React.createElement(_reactNativeSvg.Circle, {
  cx: "39.5",
  cy: "27.5",
  r: "4",
  fill: "#80DFFF",
  stroke: "#fff",
  strokeWidth: "2"
}), /*#__PURE__*/React.createElement(_reactNativeSvg.Circle, {
  cx: "16",
  cy: "24",
  r: "4",
  fill: "#19C6FF",
  stroke: "#fff",
  strokeWidth: "2"
})), /*#__PURE__*/React.createElement(_reactNativeSvg.Rect, {
  width: "59",
  height: "59",
  x: ".5",
  y: ".5",
  stroke: "#062B2B",
  strokeOpacity: ".1",
  rx: "29.5"
}), /*#__PURE__*/React.createElement(_reactNativeSvg.Defs, null, /*#__PURE__*/React.createElement(_reactNativeSvg.ClipPath, {
  id: "a"
}, /*#__PURE__*/React.createElement(_reactNativeSvg.Rect, {
  width: "60",
  height: "60",
  fill: "#fff",
  rx: "30"
}))));
var _default = exports.default = SvgNetwork;
