// @web3modal/ui-react-native/lib/commonjs/assets/visual/Noun.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _reactNativeSvg = _interopRequireWildcard(require("react-native-svg"));
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && Object.prototype.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
const SvgNoun = props => /*#__PURE__*/React.createElement(_reactNativeSvg.default, _extends({
  fill: "none",
  viewBox: "0 0 60 60"
}, props), /*#__PURE__*/React.createElement(_reactNativeSvg.Rect, {
  width: "60",
  height: "60",
  fill: "#794CFF",
  rx: "3"
}), /*#__PURE__*/React.createElement(_reactNativeSvg.Path, {
  fill: "#987DE8",
  stroke: "#fff",
  strokeWidth: "2",
  d: "M33 22.5v-1H16v5H8.5V36H13v-5h3v7.5h17V31h1v7.5h17v-17H34v5h-1v-4Z"
}), /*#__PURE__*/React.createElement(_reactNativeSvg.Path, {
  fill: "#fff",
  d: "M37.5 25h10v10h-10z"
}), /*#__PURE__*/React.createElement(_reactNativeSvg.Path, {
  fill: "#4019B2",
  d: "M42.5 25h5v10h-5z"
}), /*#__PURE__*/React.createElement(_reactNativeSvg.Path, {
  fill: "#fff",
  d: "M19.5 25h10v10h-10z"
}), /*#__PURE__*/React.createElement(_reactNativeSvg.Path, {
  fill: "#4019B2",
  d: "M24.5 25h5v10h-5z"
}), /*#__PURE__*/React.createElement(_reactNativeSvg.Path, {
  fill: "#fff",
  d: "M12 30.5h4V37h-4v-6.5Z"
}), /*#__PURE__*/React.createElement(_reactNativeSvg.Rect, {
  width: "59",
  height: "59",
  x: ".5",
  y: ".5",
  stroke: "#fff",
  strokeOpacity: ".1",
  rx: "2.5"
}));
var _default = exports.default = SvgNoun;
