// @web3modal/ui-react-native/lib/module/composites/wui-connect-button/index.js
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
import { useRef } from 'react';
import { Animated, Pressable } from 'react-native';
import { Text } from '../../components/wui-text';
import { useTheme } from '../../hooks/useTheme';
import styles, { getThemedButtonStyle, getThemedTextStyle } from './styles';
import { LoadingSpinner } from '../../components/wui-loading-spinner';
const AnimatedPressable = Animated.createAnimatedComponent(Pressable);
export function ConnectButton({
  children,
  size = 'md',
  onPress,
  style,
  loading,
  ...rest
}) {
  const Theme = useTheme();
  const themedTextStyle = getThemedTextStyle(Theme, loading);
  const colorAnimation = useRef(new Animated.Value(0));
  const themedNormalStyle = getThemedButtonStyle(Theme, loading, false);
  const themedPressedStyle = getThemedButtonStyle(Theme, loading, true);
  const onPressIn = () => {
    Animated.timing(colorAnimation.current, {
      toValue: 1,
      useNativeDriver: false,
      duration: 200
    }).start();
  };
  const onPressOut = () => {
    Animated.timing(colorAnimation.current, {
      toValue: 0,
      useNativeDriver: false,
      duration: 200
    }).start();
  };
  const backgroundColor = colorAnimation.current.interpolate({
    inputRange: [0, 1],
    outputRange: [themedNormalStyle.backgroundColor, themedPressedStyle.backgroundColor]
  });
  const borderColor = colorAnimation.current.interpolate({
    inputRange: [0, 1],
    outputRange: [themedNormalStyle.borderColor, themedPressedStyle.borderColor]
  });
  const loadingTemplate = () => {
    if (!loading) return null;
    return /*#__PURE__*/React.createElement(LoadingSpinner, {
      size: size,
      style: styles.loader
    });
  };
  return /*#__PURE__*/React.createElement(AnimatedPressable, _extends({
    disabled: loading,
    style: [styles.button, styles[`${size}Button`], {
      backgroundColor,
      borderColor
    }, style],
    onPressIn: onPressIn,
    onPressOut: onPressOut,
    onPress: onPress
  }, rest), loadingTemplate(), /*#__PURE__*/React.createElement(Text, {
    variant: size === 'md' ? 'paragraph-500' : 'small-600',
    style: themedTextStyle
  }, children));
}
