// @web3modal/ui-react-native/lib/commonjs/components/wui-image/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Image = Image;
var _react = require("react");
var _reactNative = require("react-native");
var _styles = _interopRequireDefault(require("./styles"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function Image({
  source,
  headers,
  style,
  ...rest
}) {
  const opacity = (0, _react.useRef)(new _reactNative.Animated.Value(0));
  const isIos = _reactNative.Platform.OS === 'ios';

  // Fade in image on load for iOS. Android does this by default.
  const onLoadEnd = () => {
    _reactNative.Animated.timing(opacity.current, {
      toValue: 1,
      duration: 150,
      useNativeDriver: true
    }).start();
  };
  return isIos ? /*#__PURE__*/React.createElement(_reactNative.Animated.Image, _extends({
    source: {
      uri: source,
      headers
    },
    onLoadEnd: onLoadEnd,
    style: [_styles.default.image, {
      opacity: opacity.current
    }, style]
  }, rest)) : /*#__PURE__*/React.createElement(_reactNative.Image, _extends({
    source: {
      uri: source,
      headers
    },
    style: [_styles.default.image, style]
  }, rest));
}
