// @web3modal/scaffold-react-native/lib/commonjs/views/w3m-what-is-a-wallet-view/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.WhatIsAWalletView = WhatIsAWalletView;
var _reactNative = require("react-native");
var _uiReactNative = require("@web3modal/ui-react-native");
var _coreReactNative = require("@web3modal/core-react-native");
var _useCustomDimensions = require("../../hooks/useCustomDimensions");
var _styles = _interopRequireDefault(require("./styles"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function WhatIsAWalletView() {
  const {
    padding
  } = (0, _useCustomDimensions.useCustomDimensions)();
  const onGetWalletPress = () => {
    _coreReactNative.RouterController.push('GetWallet');
    _coreReactNative.EventsController.sendEvent({
      type: 'track',
      event: 'CLICK_GET_WALLET'
    });
  };
  return /*#__PURE__*/React.createElement(_reactNative.ScrollView, {
    bounces: false,
    fadingEdgeLength: 20,
    style: {
      paddingHorizontal: padding
    }
  }, /*#__PURE__*/React.createElement(_uiReactNative.FlexView, {
    alignItems: "center",
    padding: ['l', '4xl', '3xl', '4xl']
  }, /*#__PURE__*/React.createElement(_uiReactNative.FlexView, {
    flexDirection: "row",
    padding: ['0', '0', 'xs', '0']
  }, /*#__PURE__*/React.createElement(_uiReactNative.Visual, {
    name: "login"
  }), /*#__PURE__*/React.createElement(_uiReactNative.Visual, {
    name: "profile",
    style: _styles.default.visual
  }), /*#__PURE__*/React.createElement(_uiReactNative.Visual, {
    name: "lock"
  })), /*#__PURE__*/React.createElement(_uiReactNative.Text, {
    variant: "paragraph-500",
    style: _styles.default.text
  }, "One login for all of web3"), /*#__PURE__*/React.createElement(_uiReactNative.Text, {
    variant: "small-500",
    color: "fg-200",
    center: true
  }, "Log in to any app by connecting your wallet. Say goodbye to countless passwords!"), /*#__PURE__*/React.createElement(_uiReactNative.FlexView, {
    flexDirection: "row",
    padding: ['3xl', '0', 'xs', '0']
  }, /*#__PURE__*/React.createElement(_uiReactNative.Visual, {
    name: "defi"
  }), /*#__PURE__*/React.createElement(_uiReactNative.Visual, {
    name: "nft",
    style: _styles.default.visual
  }), /*#__PURE__*/React.createElement(_uiReactNative.Visual, {
    name: "eth"
  })), /*#__PURE__*/React.createElement(_uiReactNative.Text, {
    variant: "paragraph-500",
    style: _styles.default.text
  }, "A home for your digital assets"), /*#__PURE__*/React.createElement(_uiReactNative.Text, {
    variant: "small-500",
    color: "fg-200",
    center: true
  }, "A wallet lets you store, send and receive digital assets like cryptocurrencies and NFTs."), /*#__PURE__*/React.createElement(_uiReactNative.FlexView, {
    flexDirection: "row",
    padding: ['3xl', '0', 'xs', '0']
  }, /*#__PURE__*/React.createElement(_uiReactNative.Visual, {
    name: "browser"
  }), /*#__PURE__*/React.createElement(_uiReactNative.Visual, {
    name: "noun",
    style: _styles.default.visual
  }), /*#__PURE__*/React.createElement(_uiReactNative.Visual, {
    name: "dao"
  })), /*#__PURE__*/React.createElement(_uiReactNative.Text, {
    variant: "paragraph-500",
    style: _styles.default.text
  }, "Your gateway to a new web"), /*#__PURE__*/React.createElement(_uiReactNative.Text, {
    variant: "small-500",
    color: "fg-200",
    center: true
  }, "With your wallet, you can explore and interact with DeFi, NFTs, DAOs, and much more."), /*#__PURE__*/React.createElement(_uiReactNative.Button, {
    size: "sm",
    iconLeft: "walletSmall",
    style: _styles.default.getWalletButton,
    onPress: onGetWalletPress
  }, "Get a wallet")));
}
