// @web3modal/scaffold-react-native/lib/module/views/w3m-what-is-a-wallet-view/index.js
import { ScrollView } from 'react-native';
import { Button, FlexView, Text, Visual } from '@web3modal/ui-react-native';
import { EventsController, RouterController } from '@web3modal/core-react-native';
import { useCustomDimensions } from '../../hooks/useCustomDimensions';
import styles from './styles';
export function WhatIsAWalletView() {
  const {
    padding
  } = useCustomDimensions();
  const onGetWalletPress = () => {
    RouterController.push('GetWallet');
    EventsController.sendEvent({
      type: 'track',
      event: 'CLICK_GET_WALLET'
    });
  };
  return /*#__PURE__*/React.createElement(ScrollView, {
    bounces: false,
    fadingEdgeLength: 20,
    style: {
      paddingHorizontal: padding
    }
  }, /*#__PURE__*/React.createElement(FlexView, {
    alignItems: "center",
    padding: ['l', '4xl', '3xl', '4xl']
  }, /*#__PURE__*/React.createElement(FlexView, {
    flexDirection: "row",
    padding: ['0', '0', 'xs', '0']
  }, /*#__PURE__*/React.createElement(Visual, {
    name: "login"
  }), /*#__PURE__*/React.createElement(Visual, {
    name: "profile",
    style: styles.visual
  }), /*#__PURE__*/React.createElement(Visual, {
    name: "lock"
  })), /*#__PURE__*/React.createElement(Text, {
    variant: "paragraph-500",
    style: styles.text
  }, "One login for all of web3"), /*#__PURE__*/React.createElement(Text, {
    variant: "small-500",
    color: "fg-200",
    center: true
  }, "Log in to any app by connecting your wallet. Say goodbye to countless passwords!"), /*#__PURE__*/React.createElement(FlexView, {
    flexDirection: "row",
    padding: ['3xl', '0', 'xs', '0']
  }, /*#__PURE__*/React.createElement(Visual, {
    name: "defi"
  }), /*#__PURE__*/React.createElement(Visual, {
    name: "nft",
    style: styles.visual
  }), /*#__PURE__*/React.createElement(Visual, {
    name: "eth"
  })), /*#__PURE__*/React.createElement(Text, {
    variant: "paragraph-500",
    style: styles.text
  }, "A home for your digital assets"), /*#__PURE__*/React.createElement(Text, {
    variant: "small-500",
    color: "fg-200",
    center: true
  }, "A wallet lets you store, send and receive digital assets like cryptocurrencies and NFTs."), /*#__PURE__*/React.createElement(FlexView, {
    flexDirection: "row",
    padding: ['3xl', '0', 'xs', '0']
  }, /*#__PURE__*/React.createElement(Visual, {
    name: "browser"
  }), /*#__PURE__*/React.createElement(Visual, {
    name: "noun",
    style: styles.visual
  }), /*#__PURE__*/React.createElement(Visual, {
    name: "dao"
  })), /*#__PURE__*/React.createElement(Text, {
    variant: "paragraph-500",
    style: styles.text
  }, "Your gateway to a new web"), /*#__PURE__*/React.createElement(Text, {
    variant: "small-500",
    color: "fg-200",
    center: true
  }, "With your wallet, you can explore and interact with DeFi, NFTs, DAOs, and much more."), /*#__PURE__*/React.createElement(Button, {
    size: "sm",
    iconLeft: "walletSmall",
    style: styles.getWalletButton,
    onPress: onGetWalletPress
  }, "Get a wallet")));
}
