// @web3modal/ui-react-native/lib/commonjs/composites/wui-tooltip/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Tooltip = Tooltip;
var _reactNative = require("react-native");
var _wuiIcon = require("../../components/wui-icon");
var _wuiText = require("../../components/wui-text");
var _useTheme = require("../../hooks/useTheme");
var _styles = _interopRequireDefault(require("./styles"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function Tooltip({
  message,
  placement = 'top'
}) {
  const Theme = (0, _useTheme.useTheme)();
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(_reactNative.View, {
    style: [_styles.default.container, {
      backgroundColor: Theme['fg-100']
    }]
  }, /*#__PURE__*/React.createElement(_wuiText.Text, {
    color: 'bg-100',
    variant: "small-500"
  }, message), /*#__PURE__*/React.createElement(_wuiIcon.Icon, {
    name: "cursor",
    color: "fg-100",
    style: [_styles.default.base, _styles.default[placement]],
    width: 12,
    height: 4
  })));
}
