// @web3modal/scaffold-react-native/lib/commonjs/views/w3m-what-is-a-network-view/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.WhatIsNetworkView = WhatIsNetworkView;
var _reactNative = require("react-native");
var _uiReactNative = require("@web3modal/ui-react-native");
var _useCustomDimensions = require("../../hooks/useCustomDimensions");
var _styles = _interopRequireDefault(require("./styles"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function WhatIsNetworkView() {
  const {
    padding
  } = (0, _useCustomDimensions.useCustomDimensions)();
  const onLearnMorePress = () => {
    _reactNative.Linking.openURL('https://ethereum.org/en/developers/docs/networks/');
  };
  return /*#__PURE__*/React.createElement(_reactNative.ScrollView, {
    bounces: false,
    fadingEdgeLength: 20,
    style: {
      paddingHorizontal: padding
    }
  }, /*#__PURE__*/React.createElement(_uiReactNative.FlexView, {
    alignItems: "center",
    padding: ['l', '4xl', '3xl', '4xl']
  }, /*#__PURE__*/React.createElement(_uiReactNative.FlexView, {
    flexDirection: "row",
    padding: ['0', '0', 'xs', '0']
  }, /*#__PURE__*/React.createElement(_uiReactNative.Visual, {
    name: "network"
  }), /*#__PURE__*/React.createElement(_uiReactNative.Visual, {
    name: "layers",
    style: _styles.default.visual
  }), /*#__PURE__*/React.createElement(_uiReactNative.Visual, {
    name: "system"
  })), /*#__PURE__*/React.createElement(_uiReactNative.Text, {
    variant: "paragraph-500",
    style: _styles.default.text
  }, "The system\u2019s nuts and bolts"), /*#__PURE__*/React.createElement(_uiReactNative.Text, {
    variant: "small-500",
    color: "fg-200",
    center: true
  }, "A network is what brings the blockchain to life, as this technical infrastructure allows apps to access the ledger and smart contract services."), /*#__PURE__*/React.createElement(_uiReactNative.FlexView, {
    flexDirection: "row",
    padding: ['3xl', '0', 'xs', '0']
  }, /*#__PURE__*/React.createElement(_uiReactNative.Visual, {
    name: "noun"
  }), /*#__PURE__*/React.createElement(_uiReactNative.Visual, {
    name: "defiAlt",
    style: _styles.default.visual
  }), /*#__PURE__*/React.createElement(_uiReactNative.Visual, {
    name: "dao"
  })), /*#__PURE__*/React.createElement(_uiReactNative.Text, {
    variant: "paragraph-500",
    style: _styles.default.text
  }, "Designed for different uses"), /*#__PURE__*/React.createElement(_uiReactNative.Text, {
    variant: "small-500",
    color: "fg-200",
    center: true
  }, "Each network is designed differently, and may therefore suit certain apps and experiences."), /*#__PURE__*/React.createElement(_uiReactNative.Button, {
    size: "sm",
    iconRight: "externalLink",
    style: _styles.default.learnButton,
    onPress: onLearnMorePress
  }, "Learn more")));
}
