// @web3modal/ui-react-native/lib/module/composites/wui-wallet-image/index.js
import { View } from 'react-native';
import { Icon } from '../../components/wui-icon';
import { Image } from '../../components/wui-image';
import { useTheme } from '../../hooks/useTheme';
import styles from './styles';
const IconSize = {
  xs: 10,
  sm: 24,
  md: 32,
  lg: 40
};
export function WalletImage({
  imageSrc,
  imageHeaders,
  size = 'md',
  style,
  showAllWallets
}) {
  const Theme = useTheme();
  const templateVisual = () => {
    if (imageSrc) {
      return /*#__PURE__*/React.createElement(Image, {
        source: imageSrc,
        headers: imageHeaders,
        style: styles[`${size}Image`]
      });
    }
    if (showAllWallets) {
      return /*#__PURE__*/React.createElement(View, {
        style: [styles.container, styles[`${size}Image`]]
      }, /*#__PURE__*/React.createElement(Icon, {
        height: IconSize[size],
        width: IconSize[size],
        name: "allWallets",
        color: "accent-100"
      }));
    }
    return /*#__PURE__*/React.createElement(View, {
      style: [styles.container, styles[`${size}Image`]]
    }, /*#__PURE__*/React.createElement(Icon, {
      height: IconSize[size],
      width: IconSize[size],
      name: "walletPlaceholder",
      color: "fg-200"
    }));
  };
  return /*#__PURE__*/React.createElement(View, {
    style: style
  }, templateVisual(), /*#__PURE__*/React.createElement(View, {
    style: [styles.border, styles[`${size}Image`], {
      borderColor: Theme['gray-glass-010']
    }, !imageSrc && {
      backgroundColor: Theme['gray-glass-002']
    }, showAllWallets && {
      backgroundColor: `${Theme['accent-100']}1E`,
      borderColor: Theme['accent-glass-010']
    }]
  }));
}
