// @web3modal/ui-react-native/lib/commonjs/composites/wui-qr-code/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.QrCode = QrCode;
var _react = require("react");
var _reactNative = require("react-native");
var _reactNativeSvg = _interopRequireDefault(require("react-native-svg"));
var _wuiIcon = require("../../components/wui-icon");
var _wuiImage = require("../../components/wui-image");
var _wuiShimmer = require("../../components/wui-shimmer");
var _QRCodeUtil = require("../../utils/QRCodeUtil");
var _ThemeUtil = require("../../utils/ThemeUtil");
var _styles = _interopRequireDefault(require("./styles"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function QrCode({
  size,
  uri,
  imageSrc,
  testID
}) {
  const Theme = _ThemeUtil.LightTheme;
  const containerPadding = _ThemeUtil.Spacing.l;
  const qrSize = size - containerPadding * 2;
  const dots = (0, _react.useMemo)(() => uri ? _QRCodeUtil.QRCodeUtil.generate(uri, qrSize, qrSize / 4) : [], [uri, qrSize]);
  const shimmerTemplate = () => {
    return /*#__PURE__*/React.createElement(_wuiShimmer.Shimmer, {
      width: size,
      height: size,
      borderRadius: _ThemeUtil.BorderRadius.l
    });
  };
  const logoTemplate = () => {
    if (imageSrc) {
      return /*#__PURE__*/React.createElement(_wuiImage.Image, {
        source: imageSrc,
        style: [_styles.default.icon, {
          height: qrSize / 4,
          width: qrSize / 4,
          borderRadius: qrSize / 16
        }]
      });
    }
    return /*#__PURE__*/React.createElement(_wuiIcon.Icon, {
      name: "walletConnect",
      color: "accent-100",
      height: qrSize / 3.5,
      width: qrSize / 3.5,
      style: _styles.default.icon
    });
  };
  return uri ? /*#__PURE__*/React.createElement(_reactNative.View, {
    style: [_styles.default.container, {
      height: size,
      width: size,
      backgroundColor: Theme['bg-100'],
      padding: containerPadding
    }],
    testID: testID
  }, /*#__PURE__*/React.createElement(_reactNativeSvg.default, {
    height: qrSize,
    width: qrSize
  }, dots), logoTemplate()) : shimmerTemplate();
}
