// @web3modal/ui-react-native/lib/commonjs/composites/wui-connect-button/styles.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getThemedTextStyle = exports.getThemedButtonStyle = exports.default = void 0;
var _reactNative = require("react-native");
var _ThemeUtil = require("../../utils/ThemeUtil");
const getThemedButtonStyle = (theme, loading, pressed) => {
  const buttonBaseStyle = {
    borderColor: theme['gray-glass-010']
  };
  if (loading) {
    return {
      backgroundColor: theme['gray-glass-015'],
      borderColor: theme['gray-glass-005']
    };
  }
  if (pressed) {
    return {
      ...buttonBaseStyle,
      backgroundColor: theme['accent-020']
    };
  }
  return {
    ...buttonBaseStyle,
    backgroundColor: theme['accent-100']
  };
};
exports.getThemedButtonStyle = getThemedButtonStyle;
const getThemedTextStyle = (theme, loading) => {
  if (loading) {
    return {
      color: theme['accent-100']
    };
  }
  return {
    color: theme['inverse-100']
  };
};
exports.getThemedTextStyle = getThemedTextStyle;
var _default = exports.default = _reactNative.StyleSheet.create({
  button: {
    flexDirection: 'row',
    height: 40,
    paddingVertical: _ThemeUtil.Spacing['3xs'],
    paddingHorizontal: _ThemeUtil.Spacing.l,
    alignItems: 'center',
    justifyContent: 'center',
    borderRadius: _ThemeUtil.BorderRadius.s,
    borderWidth: 1
  },
  smButton: {
    height: 32
  },
  mdButton: {
    height: 40
  },
  loader: {
    marginRight: _ThemeUtil.Spacing['2xs']
  }
});
