// @web3modal/ui-react-native/lib/commonjs/composites/wui-input-text/styles.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _reactNative = require("react-native");
var _ThemeUtil = require("../../utils/ThemeUtil");
const baseStyle = {
  flexDirection: 'row',
  height: 40,
  borderRadius: _ThemeUtil.BorderRadius.xxs,
  alignItems: 'center',
  paddingHorizontal: _ThemeUtil.Spacing.xs,
  borderWidth: 1
};
var _default = exports.default = _reactNative.StyleSheet.create({
  xsContainer: {
    ...baseStyle,
    height: 50,
    width: 50
  },
  smContainer: {
    ...baseStyle
  },
  mdContainer: {
    ...baseStyle,
    height: 56,
    paddingHorizontal: _ThemeUtil.Spacing.l,
    borderRadius: _ThemeUtil.BorderRadius.xs
  },
  outerBorder: {
    flexGrow: 1,
    borderWidth: 4,
    borderRadius: _ThemeUtil.BorderRadius.xs,
    borderColor: 'transparent'
  },
  input: {
    flex: 1,
    marginHorizontal: _ThemeUtil.Spacing['4xs'],
    fontSize: 16,
    fontWeight: '400'
  },
  icon: {
    marginLeft: _ThemeUtil.Spacing['3xs'],
    marginRight: _ThemeUtil.Spacing.xs
  }
});
