// @web3modal/ui-react-native/lib/commonjs/composites/wui-account-button/styles.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _reactNative = require("react-native");
var _ThemeUtil = require("../../utils/ThemeUtil");
var _default = exports.default = _reactNative.StyleSheet.create({
  container: {
    flexDirection: 'row',
    height: 40,
    borderRadius: 100,
    borderWidth: 1,
    justifyContent: 'center',
    alignItems: 'center',
    paddingHorizontal: _ThemeUtil.Spacing['3xs']
  },
  image: {
    height: 24,
    width: 24,
    borderRadius: 100,
    borderWidth: 2
  },
  avatarPlaceholder: {
    borderWidth: 0
  },
  networkContainer: {
    flexDirection: 'row',
    height: 32,
    paddingLeft: _ThemeUtil.Spacing['3xs'],
    paddingRight: _ThemeUtil.Spacing.xs,
    justifyContent: 'center',
    alignItems: 'center'
  },
  addressContainer: {
    flexDirection: 'row',
    height: 32,
    justifyContent: 'center',
    alignItems: 'center',
    paddingLeft: _ThemeUtil.Spacing['3xs'],
    paddingRight: _ThemeUtil.Spacing.xs,
    borderRadius: 100,
    borderWidth: 1
  },
  address: {
    marginLeft: _ThemeUtil.Spacing['3xs']
  },
  balance: {
    marginLeft: _ThemeUtil.Spacing['3xs']
  }
});
