// @web3modal/scaffold-react-native/lib/module/partials/w3m-connecting-web/index.js
import { useSnapshot } from 'valtio';
import { useCallback } from 'react';
import { Linking, ScrollView } from 'react-native';
import { RouterController, ApiController, AssetUtil, ConnectionController, CoreHelperUtil, OptionsController, EventsController } from '@web3modal/core-react-native';
import { Button, FlexView, LoadingThumbnail, Text, WalletImage, Link, IconBox } from '@web3modal/ui-react-native';
import styles from './styles';
export function ConnectingWeb({
  onCopyUri
}) {
  const {
    data
  } = useSnapshot(RouterController.state);
  const {
    wcUri,
    wcError
  } = useSnapshot(ConnectionController.state);
  const showCopy = OptionsController.isClipboardAvailable();
  const onConnect = useCallback(async () => {
    try {
      const {
        name,
        webapp_link
      } = data?.wallet ?? {};
      if (name && webapp_link && wcUri) {
        ConnectionController.setWcError(false);
        const {
          redirect,
          href
        } = CoreHelperUtil.formatUniversalUrl(webapp_link, wcUri);
        ConnectionController.setWcLinking({
          name,
          href
        });
        ConnectionController.setPressedWallet(data?.wallet);
        await Linking.openURL(redirect);
        await ConnectionController.state.wcPromise;
        EventsController.sendEvent({
          type: 'track',
          event: 'CONNECT_SUCCESS',
          properties: {
            method: 'web',
            name: data?.wallet?.name ?? 'Unknown'
          }
        });
      }
    } catch {}
  }, [data?.wallet, wcUri]);
  const textTemplate = () => {
    const walletName = data?.wallet?.name ?? 'Wallet';
    if (wcError) {
      return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(Text, {
        variant: "paragraph-500",
        color: "error-100",
        style: styles.mainText
      }, "Connection declined"), /*#__PURE__*/React.createElement(Text, {
        center: true,
        variant: "small-400",
        color: "fg-200",
        style: styles.descriptionText
      }, "Connection can be declined if a previous request is still active"));
    }
    return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(Text, {
      variant: "paragraph-500",
      style: styles.mainText
    }, `Continue in ${walletName}`), /*#__PURE__*/React.createElement(Text, {
      center: true,
      variant: "small-400",
      color: "fg-200",
      style: styles.descriptionText
    }, "Open and continue in a browser tab"));
  };
  return /*#__PURE__*/React.createElement(ScrollView, {
    bounces: false,
    fadingEdgeLength: 20
  }, /*#__PURE__*/React.createElement(FlexView, {
    alignItems: "center",
    padding: ['2xl', 'm', '3xl', 'm']
  }, /*#__PURE__*/React.createElement(LoadingThumbnail, {
    paused: wcError
  }, /*#__PURE__*/React.createElement(WalletImage, {
    size: "lg",
    imageSrc: AssetUtil.getWalletImage(data?.wallet),
    imageHeaders: ApiController._getApiHeaders()
  }), wcError && /*#__PURE__*/React.createElement(IconBox, {
    icon: 'close',
    border: true,
    background: true,
    backgroundColor: "icon-box-bg-error-100",
    size: "sm",
    iconColor: "error-100",
    style: styles.errorIcon
  })), textTemplate(), /*#__PURE__*/React.createElement(Button, {
    variant: "accent",
    iconRight: "externalLink",
    style: styles.openButton,
    onPress: onConnect
  }, "Open"), showCopy && /*#__PURE__*/React.createElement(Link, {
    iconLeft: "copySmall",
    color: "fg-200",
    style: styles.copyButton,
    onPress: () => onCopyUri(wcUri)
  }, "Copy link")));
}
