// @web3modal/ui-react-native/lib/commonjs/composites/wui-avatar/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Avatar = Avatar;
var _reactNative = require("react-native");
var _reactNativeSvg = _interopRequireWildcard(require("react-native-svg"));
var _wuiImage = require("../../components/wui-image");
var _useTheme = require("../../hooks/useTheme");
var _UiUtil = require("../../utils/UiUtil");
var _styles = _interopRequireDefault(require("./styles"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && Object.prototype.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function Avatar({
  imageSrc,
  address,
  style,
  size = 64,
  borderWidth = 8
}) {
  const Theme = (0, _useTheme.useTheme)();
  const colors = _UiUtil.UiUtil.generateAvatarColors(address);
  const containerSize = size + borderWidth * 2;
  if (imageSrc) {
    return /*#__PURE__*/React.createElement(_reactNative.View, {
      style: [_styles.default.image, {
        height: containerSize,
        width: containerSize,
        borderColor: Theme['gray-glass-005'],
        borderWidth
      }, style]
    }, /*#__PURE__*/React.createElement(_wuiImage.Image, {
      source: imageSrc,
      style: [_styles.default.image, {
        height: size,
        width: size
      }]
    }));
  }
  return /*#__PURE__*/React.createElement(_reactNativeSvg.default, {
    width: containerSize,
    height: containerSize,
    fill: "none",
    style: style
  }, /*#__PURE__*/React.createElement(_reactNativeSvg.Defs, null, /*#__PURE__*/React.createElement(_reactNativeSvg.RadialGradient, {
    id: "radial",
    cx: "0",
    cy: "0",
    r: "1",
    gradientUnits: "userSpaceOnUse",
    gradientTransform: `translate(${size * 0.75} ${size * 0.35}) scale(${size * 0.75})`
  }, /*#__PURE__*/React.createElement(_reactNativeSvg.Stop, {
    offset: "0.52%",
    stopColor: "#FFFFFF"
  }), /*#__PURE__*/React.createElement(_reactNativeSvg.Stop, {
    offset: "31.25%",
    stopColor: colors[0]
  }), /*#__PURE__*/React.createElement(_reactNativeSvg.Stop, {
    offset: "51.56%",
    stopColor: colors[1]
  }), /*#__PURE__*/React.createElement(_reactNativeSvg.Stop, {
    offset: "65.63%",
    stopColor: colors[2]
  }), /*#__PURE__*/React.createElement(_reactNativeSvg.Stop, {
    offset: "82.29%",
    stopColor: colors[3]
  }), /*#__PURE__*/React.createElement(_reactNativeSvg.Stop, {
    offset: "100%",
    stopColor: colors[4]
  }))), /*#__PURE__*/React.createElement(_reactNativeSvg.Circle, {
    cx: "50%",
    cy: "50%",
    r: size / 2,
    fill: "url(#radial)"
  }), /*#__PURE__*/React.createElement(_reactNativeSvg.Circle, {
    cx: "50%",
    cy: "50%",
    r: (size + borderWidth) / 2,
    stroke: Theme['gray-glass-005'],
    strokeWidth: borderWidth
  }));
}
