// @web3modal/ui-react-native/lib/commonjs/composites/wui-search-bar/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SearchBar = SearchBar;
var _react = require("react");
var _wuiInputElement = require("../wui-input-element");
var _wuiInputText = require("../wui-input-text");
var _ThemeUtil = require("../../utils/ThemeUtil");
function SearchBar({
  placeholder = 'Search wallet',
  onSubmitEditing,
  onChangeText,
  inputStyle,
  testID
}) {
  const [showClear, setShowClear] = (0, _react.useState)(false);
  const inputRef = (0, _react.useRef)(null);
  const handleChangeText = text => {
    setShowClear(!!text?.length);
    onChangeText?.(text);
  };
  return /*#__PURE__*/React.createElement(_wuiInputText.InputText, {
    ref: inputRef,
    placeholder: placeholder,
    onChangeText: handleChangeText,
    onSubmitEditing: onSubmitEditing,
    icon: "search",
    inputStyle: inputStyle,
    returnKeyType: "search",
    disableFullscreenUI: true,
    testID: testID
  }, showClear && /*#__PURE__*/React.createElement(_wuiInputElement.InputElement, {
    icon: "close",
    style: {
      marginRight: _ThemeUtil.Spacing['4xs']
    },
    onPress: () => {
      inputRef.current?.clear();
      inputRef.current?.focus();
      handleChangeText('');
    }
  }));
}
