// @web3modal/ui-react-native/lib/module/composites/wui-network-image/index.js
import { Path, Svg, Image, Defs, Pattern } from 'react-native-svg';
import { useTheme } from '../../hooks/useTheme';
import { PathLg, PathNormal } from './styles';
export function NetworkImage({
  imageSrc,
  imageHeaders,
  disabled,
  selected,
  size = 'md'
}) {
  const Theme = useTheme();
  const isLg = size === 'lg';
  const svgWidth = isLg ? 96 : 56;
  const svgHeight = isLg ? 96 : 56;
  const svgStroke = selected ? Theme['accent-100'] : Theme['gray-glass-010'];
  const opacity = disabled ? 0.5 : 1;
  return /*#__PURE__*/React.createElement(Svg, {
    width: svgWidth,
    height: svgHeight,
    stroke: svgStroke,
    strokeWidth: 1
  }, /*#__PURE__*/React.createElement(Defs, null, /*#__PURE__*/React.createElement(Pattern, {
    id: "image-pattern",
    x: "0",
    y: "0",
    width: "1",
    height: "1"
  }, /*#__PURE__*/React.createElement(Image, {
    height: svgHeight,
    width: svgWidth,
    x: "0",
    y: "0",
    opacity: opacity,
    href: {
      uri: imageSrc,
      headers: imageHeaders
    }
  }))), /*#__PURE__*/React.createElement(Path, {
    d: `${isLg ? PathLg : PathNormal}`,
    opacity: opacity,
    fill: imageSrc ? 'url(#image-pattern)' : Theme['gray-glass-005']
  }));
}
