// @web3modal/ui-react-native/lib/commonjs/composites/wui-logo-select/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.LogoSelect = LogoSelect;
var _reactNative = require("react-native");
var _useAnimatedValue = _interopRequireDefault(require("../../hooks/useAnimatedValue"));
var _useTheme = require("../../hooks/useTheme");
var _wuiLogo = require("../wui-logo");
var _styles = _interopRequireDefault(require("./styles"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
const AnimatedPressable = _reactNative.Animated.createAnimatedComponent(_reactNative.Pressable);
function LogoSelect({
  logo,
  disabled
}) {
  const Theme = (0, _useTheme.useTheme)();
  const {
    animatedValue,
    setStartValue,
    setEndValue
  } = (0, _useAnimatedValue.default)(Theme['gray-glass-005'], Theme['gray-glass-010']);
  return /*#__PURE__*/React.createElement(AnimatedPressable, {
    onPressIn: setEndValue,
    onPressOut: setStartValue,
    style: [_styles.default.box, {
      backgroundColor: animatedValue
    }],
    disabled: disabled
  }, /*#__PURE__*/React.createElement(_wuiLogo.Logo, {
    logo: logo,
    style: disabled && _styles.default.disabled
  }));
}
