// @web3modal/scaffold-react-native/lib/commonjs/partials/w3m-header/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Header = Header;
var _valtio = require("valtio");
var _coreReactNative = require("@web3modal/core-react-native");
var _uiReactNative = require("@web3modal/ui-react-native");
function Header() {
  const {
    view,
    history
  } = (0, _valtio.useSnapshot)(_coreReactNative.RouterController.state);
  const onHelpPress = () => {
    _coreReactNative.RouterController.push('WhatIsAWallet');
    _coreReactNative.EventsController.sendEvent({
      type: 'track',
      event: 'CLICK_WALLET_HELP'
    });
  };
  const headings = () => {
    const connectorName = _coreReactNative.RouterController.state.data?.connector?.name;
    const walletName = _coreReactNative.RouterController.state.data?.wallet?.name;
    const networkName = _coreReactNative.RouterController.state.data?.network?.name;
    return {
      Connect: 'Connect wallet',
      Account: undefined,
      ConnectingWalletConnect: walletName ?? 'WalletConnect',
      ConnectingExternal: connectorName ?? 'Connect Wallet',
      Networks: 'Select network',
      SwitchNetwork: networkName ?? 'Switch network',
      AllWallets: 'All wallets',
      WhatIsANetwork: 'What is a network?',
      WhatIsAWallet: 'What is a wallet?',
      GetWallet: 'Get a wallet'
    };
  };
  const header = headings()[view];
  const dynamicButtonTemplate = () => {
    const showBack = history.length > 1;
    return showBack ? /*#__PURE__*/React.createElement(_uiReactNative.IconLink, {
      icon: "chevronLeft",
      size: "md",
      onPress: _coreReactNative.RouterController.goBack,
      testID: "button-back"
    }) : /*#__PURE__*/React.createElement(_uiReactNative.IconLink, {
      icon: "helpCircle",
      size: "md",
      onPress: onHelpPress,
      testID: "button-help"
    });
  };
  if (!header) return null;
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(_uiReactNative.FlexView, {
    justifyContent: "space-between",
    flexDirection: "row",
    alignItems: "center",
    padding: ['l', 'xl', 'l', 'xl']
  }, dynamicButtonTemplate(), /*#__PURE__*/React.createElement(_uiReactNative.Text, {
    variant: "paragraph-600",
    numberOfLines: 1
  }, header), /*#__PURE__*/React.createElement(_uiReactNative.IconLink, {
    icon: "close",
    size: "md",
    onPress: _coreReactNative.ModalController.close,
    testID: "button-close"
  })), /*#__PURE__*/React.createElement(_uiReactNative.Separator, null));
}
