// @web3modal/ui-react-native/lib/module/assets/visual/Network.js
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
import Svg, { Circle, ClipPath, Defs, G, Rect } from 'react-native-svg';
const SvgNetwork = props => /*#__PURE__*/React.createElement(Svg, _extends({
  fill: "none",
  viewBox: "0 0 60 60"
}, props), /*#__PURE__*/React.createElement(G, {
  clipPath: "url(#a)"
}, /*#__PURE__*/React.createElement(Rect, {
  width: "60",
  height: "60",
  fill: "#00ACE6",
  rx: "30"
}), /*#__PURE__*/React.createElement(Circle, {
  cx: "64",
  cy: "39",
  r: "50",
  fill: "#1AC6FF",
  stroke: "#fff",
  strokeWidth: "2"
}), /*#__PURE__*/React.createElement(Circle, {
  cx: "78",
  cy: "30",
  r: "50",
  fill: "#4DD2FF",
  stroke: "#fff",
  strokeWidth: "2"
}), /*#__PURE__*/React.createElement(Circle, {
  cx: "72",
  cy: "15",
  r: "35",
  fill: "#80DFFF",
  stroke: "#fff",
  strokeWidth: "2"
}), /*#__PURE__*/React.createElement(Circle, {
  cx: "34",
  cy: "-17",
  r: "45",
  stroke: "#fff",
  strokeWidth: "2"
}), /*#__PURE__*/React.createElement(Circle, {
  cx: "34",
  cy: "-5",
  r: "50",
  stroke: "#fff",
  strokeWidth: "2"
}), /*#__PURE__*/React.createElement(Circle, {
  cx: "30",
  cy: "45",
  r: "4",
  fill: "#4DD2FF",
  stroke: "#fff",
  strokeWidth: "2"
}), /*#__PURE__*/React.createElement(Circle, {
  cx: "39.5",
  cy: "27.5",
  r: "4",
  fill: "#80DFFF",
  stroke: "#fff",
  strokeWidth: "2"
}), /*#__PURE__*/React.createElement(Circle, {
  cx: "16",
  cy: "24",
  r: "4",
  fill: "#19C6FF",
  stroke: "#fff",
  strokeWidth: "2"
})), /*#__PURE__*/React.createElement(Rect, {
  width: "59",
  height: "59",
  x: ".5",
  y: ".5",
  stroke: "#062B2B",
  strokeOpacity: ".1",
  rx: "29.5"
}), /*#__PURE__*/React.createElement(Defs, null, /*#__PURE__*/React.createElement(ClipPath, {
  id: "a"
}, /*#__PURE__*/React.createElement(Rect, {
  width: "60",
  height: "60",
  fill: "#fff",
  rx: "30"
}))));
export default SvgNetwork;
