// @web3modal/ui-react-native/lib/module/assets/visual/Noun.js
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
import Svg, { Path, Rect } from 'react-native-svg';
const SvgNoun = props => /*#__PURE__*/React.createElement(Svg, _extends({
  fill: "none",
  viewBox: "0 0 60 60"
}, props), /*#__PURE__*/React.createElement(Rect, {
  width: "60",
  height: "60",
  fill: "#794CFF",
  rx: "3"
}), /*#__PURE__*/React.createElement(Path, {
  fill: "#987DE8",
  stroke: "#fff",
  strokeWidth: "2",
  d: "M33 22.5v-1H16v5H8.5V36H13v-5h3v7.5h17V31h1v7.5h17v-17H34v5h-1v-4Z"
}), /*#__PURE__*/React.createElement(Path, {
  fill: "#fff",
  d: "M37.5 25h10v10h-10z"
}), /*#__PURE__*/React.createElement(Path, {
  fill: "#4019B2",
  d: "M42.5 25h5v10h-5z"
}), /*#__PURE__*/React.createElement(Path, {
  fill: "#fff",
  d: "M19.5 25h10v10h-10z"
}), /*#__PURE__*/React.createElement(Path, {
  fill: "#4019B2",
  d: "M24.5 25h5v10h-5z"
}), /*#__PURE__*/React.createElement(Path, {
  fill: "#fff",
  d: "M12 30.5h4V37h-4v-6.5Z"
}), /*#__PURE__*/React.createElement(Rect, {
  width: "59",
  height: "59",
  x: ".5",
  y: ".5",
  stroke: "#fff",
  strokeOpacity: ".1",
  rx: "2.5"
}));
export default SvgNoun;
