// @web3modal/ui-react-native/lib/commonjs/composites/wui-card-select/styles.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getBackgroundColor = exports.default = exports.ITEM_WIDTH = exports.ITEM_HEIGHT = void 0;
var _reactNative = require("react-native");
var _ThemeUtil = require("../../utils/ThemeUtil");
const ITEM_HEIGHT = exports.ITEM_HEIGHT = 96;
const ITEM_WIDTH = exports.ITEM_WIDTH = 76;
const getBackgroundColor = ({
  selected,
  disabled,
  pressed
}) => {
  if (disabled) return 'gray-glass-005';
  if (selected) return 'accent-glass-020';
  return pressed ? 'gray-glass-010' : 'gray-glass-002';
};
exports.getBackgroundColor = getBackgroundColor;
var _default = exports.default = _reactNative.StyleSheet.create({
  container: {
    justifyContent: 'center',
    alignItems: 'center',
    height: ITEM_HEIGHT,
    width: ITEM_WIDTH,
    borderRadius: _ThemeUtil.BorderRadius.xs
  },
  text: {
    marginTop: _ThemeUtil.Spacing.xs,
    marginHorizontal: _ThemeUtil.Spacing['2xs']
  },
  disabledImage: {
    opacity: 0.4
  },
  installedBox: {
    position: 'absolute',
    bottom: -5,
    right: -5,
    height: 24,
    width: 24,
    zIndex: 2
  }
});
