// @web3modal/ui-react-native/lib/module/components/wui-loading-spinner/index.js
import { useEffect, useRef } from 'react';
import { Animated, Easing, View } from 'react-native';
import Svg, { Circle } from 'react-native-svg';
import { useTheme } from '../../hooks/useTheme';
import { SpinnerSize } from '../../utils/ThemeUtil';
import styles from './styles';
const AnimatedSvg = Animated.createAnimatedComponent(Svg);
export function LoadingSpinner({
  color,
  style,
  size = 'lg'
}) {
  const Theme = useTheme();
  const spinValue = useRef(new Animated.Value(0));
  const stroke = color ? Theme[color] : Theme['accent-100'];
  useEffect(() => {
    const animation = Animated.timing(spinValue.current, {
      toValue: 1,
      duration: 800,
      useNativeDriver: true,
      easing: Easing.linear
    });
    const loop = Animated.loop(animation);
    loop.start();
    return () => {
      loop.stop();
    };
  }, [spinValue]);
  const spin = spinValue.current.interpolate({
    inputRange: [0, 1],
    outputRange: ['0deg', '360deg']
  });
  return /*#__PURE__*/React.createElement(View, {
    style: [styles.container, style]
  }, /*#__PURE__*/React.createElement(AnimatedSvg, {
    width: SpinnerSize[size],
    height: SpinnerSize[size],
    viewBox: "25 25 50 50",
    style: {
      transform: [{
        rotate: spin
      }]
    }
  }, /*#__PURE__*/React.createElement(Circle, {
    r: 20,
    cy: 50,
    cx: 50,
    stroke: stroke,
    strokeWidth: 4,
    fill: "transparent",
    strokeDasharray: `90 124`
  })));
}
