// @web3modal/scaffold-react-native/lib/commonjs/views/w3m-network-switch-view/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.NetworkSwitchView = NetworkSwitchView;
var _valtio = require("valtio");
var _react = require("react");
var _coreReactNative = require("@web3modal/core-react-native");
var _uiReactNative = require("@web3modal/ui-react-native");
var _styles = _interopRequireDefault(require("./styles"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function NetworkSwitchView() {
  const {
    data
  } = (0, _valtio.useSnapshot)(_coreReactNative.RouterController.state);
  const {
    recentWallets
  } = (0, _valtio.useSnapshot)(_coreReactNative.ConnectionController.state);
  const {
    caipNetwork
  } = (0, _valtio.useSnapshot)(_coreReactNative.NetworkController.state);
  const [error, setError] = (0, _react.useState)(false);
  const [showRetry, setShowRetry] = (0, _react.useState)(false);
  const network = data?.network;
  const wallet = recentWallets?.[0];
  const onSwitchNetwork = async () => {
    try {
      setError(false);
      await _coreReactNative.NetworkController.switchActiveNetwork(network);
      _coreReactNative.EventsController.sendEvent({
        type: 'track',
        event: 'SWITCH_NETWORK',
        properties: {
          network: network.id
        }
      });
    } catch {
      setError(true);
      setShowRetry(true);
    }
  };
  (0, _react.useEffect)(() => {
    onSwitchNetwork();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);
  (0, _react.useEffect)(() => {
    // Go back if network is already switched
    if (caipNetwork?.id === network?.id) {
      _coreReactNative.RouterController.goBack();
    }
  }, [caipNetwork?.id, network?.id]);
  const retryTemplate = () => {
    if (!showRetry) return null;
    return /*#__PURE__*/React.createElement(_uiReactNative.Button, {
      variant: "accent",
      iconLeft: "refresh",
      style: _styles.default.retryButton,
      iconStyle: _styles.default.retryIcon,
      onPress: onSwitchNetwork
    }, "Try again");
  };
  const textTemplate = () => {
    const walletName = wallet?.name ?? 'wallet';
    if (error) {
      return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(_uiReactNative.Text, {
        variant: "paragraph-500",
        style: _styles.default.text
      }, "Switch declined"), /*#__PURE__*/React.createElement(_uiReactNative.Text, {
        center: true,
        variant: "small-400",
        color: "fg-200",
        style: _styles.default.descriptionText
      }, "Switch can be declined if chain is not supported by a wallet or previous request is still active"));
    }
    return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(_uiReactNative.Text, {
      variant: "paragraph-500",
      style: _styles.default.text
    }, `Approve in ${walletName}`), /*#__PURE__*/React.createElement(_uiReactNative.Text, {
      center: true,
      variant: "small-400",
      color: "fg-200",
      style: _styles.default.descriptionText
    }, "Accept switch request in your wallet"));
  };
  return /*#__PURE__*/React.createElement(_uiReactNative.FlexView, {
    alignItems: "center",
    padding: ['2xl', 's', '4xl', 's']
  }, /*#__PURE__*/React.createElement(_uiReactNative.LoadingHexagon, {
    paused: error
  }, /*#__PURE__*/React.createElement(_uiReactNative.NetworkImage, {
    imageSrc: _coreReactNative.AssetUtil.getNetworkImage(network),
    imageHeaders: _coreReactNative.ApiController._getApiHeaders(),
    size: "lg"
  }), error && /*#__PURE__*/React.createElement(_uiReactNative.IconBox, {
    icon: "close",
    border: true,
    background: true,
    backgroundColor: "icon-box-bg-error-100",
    size: "sm",
    iconColor: "error-100",
    style: _styles.default.errorIcon
  })), textTemplate(), retryTemplate());
}
