// @web3modal/scaffold-react-native/lib/commonjs/modal/w3m-network-button/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.W3mNetworkButton = W3mNetworkButton;
var _valtio = require("valtio");
var _coreReactNative = require("@web3modal/core-react-native");
var _uiReactNative = require("@web3modal/ui-react-native");
function W3mNetworkButton({
  disabled,
  style
}) {
  const {
    isConnected
  } = (0, _valtio.useSnapshot)(_coreReactNative.AccountController.state);
  const {
    caipNetwork
  } = (0, _valtio.useSnapshot)(_coreReactNative.NetworkController.state);
  const onNetworkPress = () => {
    _coreReactNative.ModalController.open({
      view: 'Networks'
    });
    _coreReactNative.EventsController.sendEvent({
      type: 'track',
      event: 'CLICK_NETWORKS'
    });
  };
  return /*#__PURE__*/React.createElement(_uiReactNative.NetworkButton, {
    imageSrc: _coreReactNative.AssetUtil.getNetworkImage(caipNetwork),
    imageHeaders: _coreReactNative.ApiController._getApiHeaders(),
    disabled: disabled,
    style: style,
    onPress: onNetworkPress
  }, caipNetwork?.name ?? (isConnected ? 'Unknown Network' : 'Select Network'));
}
