// @web3modal/scaffold-react-native/lib/commonjs/partials/w3m-snackbar/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Snackbar = Snackbar;
var _valtio = require("valtio");
var _react = require("react");
var _reactNative = require("react-native");
var _coreReactNative = require("@web3modal/core-react-native");
var _uiReactNative = require("@web3modal/ui-react-native");
var _styles = _interopRequireDefault(require("./styles"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function Snackbar() {
  const {
    open,
    message,
    variant
  } = (0, _valtio.useSnapshot)(_coreReactNative.SnackController.state);
  const componentOpacity = (0, _react.useMemo)(() => new _reactNative.Animated.Value(0), []);
  (0, _react.useEffect)(() => {
    if (open) {
      _reactNative.Animated.timing(componentOpacity, {
        toValue: 1,
        duration: 150,
        useNativeDriver: true
      }).start();
      setTimeout(() => {
        _reactNative.Animated.timing(componentOpacity, {
          toValue: 0,
          duration: 300,
          useNativeDriver: true
        }).start(() => {
          _coreReactNative.SnackController.hide();
        });
      }, 2200);
    }
  }, [open, componentOpacity]);
  return /*#__PURE__*/React.createElement(_uiReactNative.Snackbar, {
    message: message,
    icon: variant === 'success' ? 'checkmark' : 'close',
    iconColor: variant === 'success' ? 'success-100' : 'error-100',
    style: [_styles.default.container, {
      opacity: componentOpacity
    }]
  });
}
