// @web3modal/ui-react-native/lib/commonjs/composites/wui-action-entry/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ActionEntry = ActionEntry;
var _reactNative = require("react-native");
var _useAnimatedValue = _interopRequireDefault(require("../../hooks/useAnimatedValue"));
var _useTheme = require("../../hooks/useTheme");
var _styles = _interopRequireDefault(require("./styles"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
const AnimatedPressable = _reactNative.Animated.createAnimatedComponent(_reactNative.Pressable);
function ActionEntry({
  children,
  onPress,
  disabled,
  style
}) {
  const Theme = (0, _useTheme.useTheme)();
  const {
    animatedValue,
    setStartValue,
    setEndValue
  } = (0, _useAnimatedValue.default)(Theme['gray-glass-002'], Theme['gray-glass-010']);
  const isPressable = !!onPress;
  return /*#__PURE__*/React.createElement(AnimatedPressable, {
    style: [_styles.default.container, {
      backgroundColor: disabled ? Theme['gray-glass-010'] : animatedValue
    }, style],
    disabled: disabled,
    onPress: onPress,
    onPressIn: isPressable ? setEndValue : undefined,
    onPressOut: isPressable ? setStartValue : undefined
  }, children);
}
