// @web3modal/scaffold-react-native/lib/module/partials/w3m-connecting-mobile/index.js
import { useSnapshot } from 'valtio';
import { useCallback, useEffect, useState } from 'react';
import { Linking, Platform, ScrollView } from 'react-native';
import { RouterController, ApiController, AssetUtil, ConnectionController, CoreHelperUtil, OptionsController, EventsController } from '@web3modal/core-react-native';
import { Button, FlexView, LoadingThumbnail, Text, WalletImage, Link, IconBox, ActionEntry } from '@web3modal/ui-react-native';
import { useCustomDimensions } from '../../hooks/useCustomDimensions';
import styles from './styles';
export function ConnectingMobile({
  onRetry,
  onCopyUri,
  isInstalled
}) {
  const {
    data
  } = useSnapshot(RouterController.state);
  const {
    maxWidth: width
  } = useCustomDimensions();
  const {
    wcUri,
    wcError
  } = useSnapshot(ConnectionController.state);
  const [linkingError, setLinkingError] = useState(false);
  const [isRetrying, setIsRetrying] = useState(false);
  const [ready, setReady] = useState(false);
  const showCopy = OptionsController.isClipboardAvailable() && !linkingError;
  const storeUrl = Platform.select({
    ios: data?.wallet?.app_store,
    android: data?.wallet?.play_store
  });
  const onRetryPress = () => {
    onRetry();
    setIsRetrying(true);
  };
  const onStorePress = () => {
    if (storeUrl) {
      Linking.openURL(storeUrl);
    }
  };
  const onConnect = useCallback(async () => {
    try {
      const {
        name,
        mobile_link
      } = data?.wallet ?? {};
      if (name && mobile_link && wcUri) {
        setLinkingError(false);
        ConnectionController.setWcError(false);
        const {
          redirect,
          href
        } = CoreHelperUtil.formatNativeUrl(mobile_link, wcUri);
        ConnectionController.setWcLinking({
          name,
          href
        });
        ConnectionController.setPressedWallet(data?.wallet);
        await Linking.openURL(redirect);
        await ConnectionController.state.wcPromise;
        EventsController.sendEvent({
          type: 'track',
          event: 'CONNECT_SUCCESS',
          properties: {
            method: 'mobile',
            name: data?.wallet?.name ?? 'Unknown'
          }
        });
      }
    } catch (error) {
      setLinkingError(true);
    }
  }, [data?.wallet, wcUri]);
  const textTemplate = () => {
    const walletName = data?.wallet?.name ?? 'Wallet';
    if (linkingError) {
      return /*#__PURE__*/React.createElement(FlexView, {
        padding: ['3xs', '2xl', '0', '2xl'],
        alignItems: "center",
        style: styles.textContainer
      }, /*#__PURE__*/React.createElement(Text, {
        variant: "paragraph-500"
      }, "App not installed"));
    } else if (wcError) {
      return /*#__PURE__*/React.createElement(FlexView, {
        padding: ['3xs', '2xl', '0', '2xl'],
        alignItems: "center",
        style: styles.textContainer
      }, /*#__PURE__*/React.createElement(Text, {
        variant: "paragraph-500",
        color: "error-100"
      }, "Connection declined"), /*#__PURE__*/React.createElement(Text, {
        center: true,
        variant: "small-400",
        color: "fg-200",
        style: styles.descriptionText
      }, "Connection can be declined if a previous request is still active"));
    }
    return /*#__PURE__*/React.createElement(FlexView, {
      padding: ['3xs', '2xl', '0', '2xl'],
      alignItems: "center",
      style: styles.textContainer
    }, /*#__PURE__*/React.createElement(Text, {
      variant: "paragraph-500"
    }, `Continue in ${walletName}`), /*#__PURE__*/React.createElement(Text, {
      center: true,
      variant: "small-400",
      color: "fg-200",
      style: styles.descriptionText
    }, "Accept connection request in the wallet"));
  };
  const storeTemplate = () => {
    if (!storeUrl || isInstalled) return null;
    return /*#__PURE__*/React.createElement(ActionEntry, {
      style: styles.storeButton
    }, /*#__PURE__*/React.createElement(Text, {
      numberOfLines: 1,
      variant: "paragraph-500",
      color: "fg-200"
    }, `Don't have ${data?.wallet?.name}?`), /*#__PURE__*/React.createElement(Button, {
      variant: "accent",
      iconRight: "chevronRightSmall",
      onPress: onStorePress,
      size: "sm",
      hitSlop: 20
    }, "Get"));
  };
  useEffect(() => {
    // First connection
    if (!ready && wcUri) {
      setReady(true);
      onConnect();
    }
  }, [ready, wcUri, onConnect]);
  useEffect(() => {
    if (isRetrying) {
      setIsRetrying(false);
      onConnect();
    }
  }, [wcUri, isRetrying, onConnect]);
  return /*#__PURE__*/React.createElement(ScrollView, {
    bounces: false,
    fadingEdgeLength: 20,
    contentContainerStyle: styles.container
  }, /*#__PURE__*/React.createElement(FlexView, {
    alignItems: "center",
    alignSelf: "center",
    padding: ['2xl', 'l', '0', 'l'],
    style: {
      width
    }
  }, /*#__PURE__*/React.createElement(LoadingThumbnail, {
    paused: linkingError || wcError
  }, /*#__PURE__*/React.createElement(WalletImage, {
    size: "lg",
    imageSrc: AssetUtil.getWalletImage(data?.wallet),
    imageHeaders: ApiController._getApiHeaders()
  }), wcError && /*#__PURE__*/React.createElement(IconBox, {
    icon: 'close',
    border: true,
    background: true,
    backgroundColor: "icon-box-bg-error-100",
    size: "sm",
    iconColor: "error-100",
    style: styles.errorIcon
  })), textTemplate(), !linkingError && /*#__PURE__*/React.createElement(Button, {
    variant: "accent",
    iconLeft: "refresh",
    style: styles.retryButton,
    iconStyle: styles.retryIcon,
    onPress: onRetryPress
  }, "Try again")), showCopy && /*#__PURE__*/React.createElement(Link, {
    iconLeft: "copySmall",
    color: "fg-200",
    style: styles.copyButton,
    onPress: () => onCopyUri(wcUri)
  }, "Copy link"), storeTemplate());
}
