// @web3modal/scaffold-react-native/lib/commonjs/client.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Web3ModalScaffold = void 0;
require("./config/animations");
var _coreReactNative = require("@web3modal/core-react-native");
// -- Types ---------------------------------------------------------------------

// -- Client --------------------------------------------------------------------
class Web3ModalScaffold {
  constructor(options) {
    this.initControllers(options);
  }

  // -- Public -------------------------------------------------------------------
  async open(options) {
    _coreReactNative.ModalController.open(options);
  }
  async close() {
    _coreReactNative.ModalController.close();
  }
  getThemeMode() {
    return _coreReactNative.ThemeController.state.themeMode;
  }
  getThemeVariables() {
    return _coreReactNative.ThemeController.state.themeVariables;
  }
  setThemeMode(themeMode) {
    _coreReactNative.ThemeController.setThemeMode(themeMode);
  }
  setThemeVariables(themeVariables) {
    _coreReactNative.ThemeController.setThemeVariables(themeVariables);
  }
  subscribeTheme(callback) {
    return _coreReactNative.ThemeController.subscribe(callback);
  }
  getState() {
    return {
      ..._coreReactNative.PublicStateController.state
    };
  }
  subscribeState(callback) {
    return _coreReactNative.PublicStateController.subscribe(callback);
  }
  subscribeStateKey(key, callback) {
    return _coreReactNative.PublicStateController.subscribeKey(key, callback);
  }
  subscribeConnection(callback) {
    return _coreReactNative.AccountController.subscribeConnection(callback);
  }
  setLoading(loading) {
    _coreReactNative.ModalController.setLoading(loading);
  }
  getEvent() {
    return {
      ..._coreReactNative.EventsController.state
    };
  }
  subscribeEvents(callback) {
    return _coreReactNative.EventsController.subscribe(callback);
  }

  // -- Protected ----------------------------------------------------------------
  setIsConnected = isConnected => {
    _coreReactNative.AccountController.setIsConnected(isConnected);
  };
  setCaipAddress = caipAddress => {
    _coreReactNative.AccountController.setCaipAddress(caipAddress);
  };
  setBalance = (balance, balanceSymbol) => {
    _coreReactNative.AccountController.setBalance(balance, balanceSymbol);
  };
  setProfileName = profileName => {
    _coreReactNative.AccountController.setProfileName(profileName);
  };
  setProfileImage = profileImage => {
    _coreReactNative.AccountController.setProfileImage(profileImage);
  };
  resetAccount = () => {
    _coreReactNative.AccountController.resetAccount();
  };
  setCaipNetwork = caipNetwork => {
    _coreReactNative.NetworkController.setCaipNetwork(caipNetwork);
  };
  getCaipNetwork = () => _coreReactNative.NetworkController.state.caipNetwork;
  setRequestedCaipNetworks = requestedCaipNetworks => {
    _coreReactNative.NetworkController.setRequestedCaipNetworks(requestedCaipNetworks);
  };
  getApprovedCaipNetworksData = () => _coreReactNative.NetworkController.getApprovedCaipNetworksData();
  resetNetwork = () => {
    _coreReactNative.NetworkController.resetNetwork();
  };
  setConnectors = connectors => {
    _coreReactNative.ConnectorController.setConnectors(connectors);
  };
  addConnector = connector => {
    _coreReactNative.ConnectorController.addConnector(connector);
  };
  getConnectors = () => _coreReactNative.ConnectorController.getConnectors();
  resetWcConnection = () => {
    _coreReactNative.ConnectionController.resetWcConnection();
  };
  fetchIdentity = request => _coreReactNative.BlockchainApiController.fetchIdentity(request);
  setAddressExplorerUrl = addressExplorerUrl => {
    _coreReactNative.AccountController.setAddressExplorerUrl(addressExplorerUrl);
  };

  // -- Private ------------------------------------------------------------------
  initControllers(options) {
    this.initRecentWallets(options);
    _coreReactNative.NetworkController.setClient(options.networkControllerClient);
    _coreReactNative.NetworkController.setDefaultCaipNetwork(options.defaultChain);
    _coreReactNative.OptionsController.setProjectId(options.projectId);
    _coreReactNative.OptionsController.setIncludeWalletIds(options.includeWalletIds);
    _coreReactNative.OptionsController.setExcludeWalletIds(options.excludeWalletIds);
    _coreReactNative.OptionsController.setFeaturedWalletIds(options.featuredWalletIds);
    _coreReactNative.OptionsController.setTokens(options.tokens);
    _coreReactNative.OptionsController.setCustomWallets(options.customWallets);
    _coreReactNative.OptionsController.setEnableAnalytics(options.enableAnalytics);
    _coreReactNative.OptionsController.setSdkVersion(options._sdkVersion);
    if (options.clipboardClient) {
      _coreReactNative.OptionsController.setClipboardClient(options.clipboardClient);
    }
    _coreReactNative.ConnectionController.setClient(options.connectionControllerClient);
    if (options.themeMode) {
      _coreReactNative.ThemeController.setThemeMode(options.themeMode);
    }
    if (options.themeVariables) {
      _coreReactNative.ThemeController.setThemeVariables(options.themeVariables);
    }
  }
  async initRecentWallets(options) {
    const wallets = await _coreReactNative.StorageUtil.getRecentWallets();
    const filteredWallets = wallets.filter(wallet => {
      const {
        includeWalletIds,
        excludeWalletIds
      } = options;
      if (includeWalletIds) {
        return includeWalletIds.includes(wallet.id);
      }
      if (excludeWalletIds) {
        return !excludeWalletIds.includes(wallet.id);
      }
      return true;
    });
    _coreReactNative.ConnectionController.setRecentWallets(filteredWallets);
  }
}
exports.Web3ModalScaffold = Web3ModalScaffold;
