// @web3modal/ui-react-native/lib/module/components/wui-image/index.js
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
import { useRef } from 'react';
import { Animated, Image as NativeImage, Platform } from 'react-native';
import styles from './styles';
export function Image({
  source,
  headers,
  style,
  ...rest
}) {
  const opacity = useRef(new Animated.Value(0));
  const isIos = Platform.OS === 'ios';

  // Fade in image on load for iOS. Android does this by default.
  const onLoadEnd = () => {
    Animated.timing(opacity.current, {
      toValue: 1,
      duration: 150,
      useNativeDriver: true
    }).start();
  };
  return isIos ? /*#__PURE__*/React.createElement(Animated.Image, _extends({
    source: {
      uri: source,
      headers
    },
    onLoadEnd: onLoadEnd,
    style: [styles.image, {
      opacity: opacity.current
    }, style]
  }, rest)) : /*#__PURE__*/React.createElement(NativeImage, _extends({
    source: {
      uri: source,
      headers
    },
    style: [styles.image, style]
  }, rest));
}
