// @web3modal/ui-react-native/lib/commonjs/composites/wui-network-button/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.NetworkButton = NetworkButton;
var _reactNative = require("react-native");
var _wuiImage = require("../../components/wui-image");
var _wuiText = require("../../components/wui-text");
var _useTheme = require("../../hooks/useTheme");
var _wuiIconBox = require("../wui-icon-box");
var _styles = _interopRequireDefault(require("./styles"));
var _useAnimatedValue = _interopRequireDefault(require("../../hooks/useAnimatedValue"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
const AnimatedPressable = _reactNative.Animated.createAnimatedComponent(_reactNative.Pressable);
function NetworkButton({
  imageSrc,
  imageHeaders,
  disabled,
  onPress,
  style,
  children
}) {
  const Theme = (0, _useTheme.useTheme)();
  const textColor = disabled ? 'fg-300' : 'fg-100';
  const borderColor = disabled ? 'gray-glass-005' : 'gray-glass-010';
  const {
    animatedValue,
    setStartValue,
    setEndValue
  } = (0, _useAnimatedValue.default)(Theme['gray-glass-002'], Theme['gray-glass-010']);
  const backgroundColor = disabled ? Theme['gray-glass-020'] : animatedValue;
  return /*#__PURE__*/React.createElement(AnimatedPressable, {
    style: [_styles.default.container, {
      backgroundColor,
      borderColor: Theme[borderColor]
    }, style],
    onPress: onPress,
    onPressIn: setEndValue,
    onPressOut: setStartValue,
    disabled: disabled
  }, imageSrc ? /*#__PURE__*/React.createElement(_wuiImage.Image, {
    style: [_styles.default.image, {
      borderColor: Theme[borderColor]
    }, disabled && _styles.default.imageDisabled],
    source: imageSrc,
    headers: imageHeaders
  }) : /*#__PURE__*/React.createElement(_wuiIconBox.IconBox, {
    icon: "networkPlaceholder",
    background: true,
    iconColor: textColor,
    size: "sm"
  }), /*#__PURE__*/React.createElement(_wuiText.Text, {
    style: _styles.default.text,
    variant: "paragraph-600",
    color: textColor
  }, children));
}
