// @web3modal/scaffold-react-native/lib/module/views/w3m-account-view/index.js
import { useSnapshot } from 'valtio';
import { Linking, ScrollView } from 'react-native';
import { AccountController, ApiController, AssetUtil, ConnectionController, CoreHelperUtil, EventsController, ModalController, NetworkController, OptionsController, RouterController, SnackController } from '@web3modal/core-react-native';
import { Avatar, Button, FlexView, IconLink, Text, UiUtil, Spacing, ListItem } from '@web3modal/ui-react-native';
import { useState } from 'react';
import { useCustomDimensions } from '../../hooks/useCustomDimensions';
import styles from './styles';
export function AccountView() {
  const {
    address,
    profileName,
    profileImage,
    balance,
    balanceSymbol,
    addressExplorerUrl
  } = useSnapshot(AccountController.state);
  const [disconnecting, setDisconnecting] = useState(false);
  const {
    caipNetwork
  } = useSnapshot(NetworkController.state);
  const networkImage = AssetUtil.getNetworkImage(caipNetwork);
  const showCopy = OptionsController.isClipboardAvailable();
  const {
    padding
  } = useCustomDimensions();
  async function onDisconnect() {
    try {
      setDisconnecting(true);
      await ConnectionController.disconnect();
      AccountController.setIsConnected(false);
      ModalController.close();
      setDisconnecting(false);
      EventsController.sendEvent({
        type: 'track',
        event: 'DISCONNECT_SUCCESS'
      });
    } catch (error) {
      EventsController.sendEvent({
        type: 'track',
        event: 'DISCONNECT_ERROR'
      });
    }
  }
  const onExplorerPress = () => {
    if (addressExplorerUrl) {
      Linking.openURL(addressExplorerUrl);
    }
  };
  const onCopyAddress = () => {
    if (address) {
      OptionsController.copyToClipboard(profileName ?? address);
      SnackController.showSuccess('Address copied');
    }
  };
  const onNetworkPress = () => {
    RouterController.push('Networks');
    EventsController.sendEvent({
      type: 'track',
      event: 'CLICK_NETWORKS'
    });
  };
  const addressExplorerTemplate = () => {
    if (!addressExplorerUrl) return null;
    return /*#__PURE__*/React.createElement(Button, {
      size: "sm",
      variant: "shade",
      iconLeft: "compass",
      iconRight: "externalLink",
      onPress: onExplorerPress,
      style: {
        marginVertical: Spacing.s
      }
    }, "Block Explorer");
  };
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(IconLink, {
    icon: "close",
    style: styles.closeIcon,
    onPress: ModalController.close
  }), /*#__PURE__*/React.createElement(ScrollView, {
    bounces: false,
    fadingEdgeLength: 20,
    style: {
      paddingHorizontal: padding
    }
  }, /*#__PURE__*/React.createElement(FlexView, {
    alignItems: "center",
    padding: ['3xl', 's', '3xl', 's']
  }, /*#__PURE__*/React.createElement(Avatar, {
    imageSrc: profileImage,
    address: profileName ?? address
  }), /*#__PURE__*/React.createElement(FlexView, {
    flexDirection: "row",
    alignItems: "center",
    margin: ['s', '0', '0', '0']
  }, /*#__PURE__*/React.createElement(Text, {
    variant: "medium-title-600"
  }, profileName ? UiUtil.getTruncateString({
    string: profileName,
    charsStart: 20,
    charsEnd: 0,
    truncate: 'end'
  }) : UiUtil.getTruncateString({
    string: address ?? '',
    charsStart: 4,
    charsEnd: 6,
    truncate: 'middle'
  })), showCopy && /*#__PURE__*/React.createElement(IconLink, {
    icon: "copy",
    size: "md",
    iconColor: "fg-250",
    onPress: onCopyAddress,
    style: styles.copyButton
  })), balance && /*#__PURE__*/React.createElement(Text, {
    variant: "paragraph-400",
    color: "fg-200"
  }, CoreHelperUtil.formatBalance(balance, balanceSymbol)), addressExplorerTemplate(), /*#__PURE__*/React.createElement(FlexView, {
    margin: ['s', '0', '0', '0']
  }, /*#__PURE__*/React.createElement(ListItem, {
    variant: networkImage ? 'image' : 'icon',
    chevron: true,
    icon: "networkPlaceholder",
    iconVariant: "overlay",
    imageSrc: networkImage,
    imageHeaders: ApiController._getApiHeaders(),
    onPress: onNetworkPress,
    testID: "button-network"
  }, /*#__PURE__*/React.createElement(Text, {
    numberOfLines: 1,
    color: "fg-100"
  }, caipNetwork?.name)), /*#__PURE__*/React.createElement(ListItem, {
    variant: "icon",
    icon: "disconnect",
    iconVariant: "overlay",
    onPress: onDisconnect,
    loading: disconnecting,
    style: styles.disconnectButton,
    testID: "button-disconnect"
  }, /*#__PURE__*/React.createElement(Text, {
    color: "fg-200"
  }, "Disconnect"))))));
}
