// @web3modal/ui-react-native/lib/module/components/wui-visual/index.js
import BrowserSvg from '../../assets/visual/Browser';
import DaoSvg from '../../assets/visual/Dao';
import DefiSvg from '../../assets/visual/Defi';
import DefiAltSvg from '../../assets/visual/DefiAlt';
import EthSvg from '../../assets/visual/Eth';
import LayersSvg from '../../assets/visual/Layers';
import LockSvg from '../../assets/visual/Lock';
import LoginSvg from '../../assets/visual/Login';
import NetworkSvg from '../../assets/visual/Network';
import NftSvg from '../../assets/visual/Nft';
import NounSvg from '../../assets/visual/Noun';
import ProfileSvg from '../../assets/visual/Profile';
import SystemSvg from '../../assets/visual/System';
const svgOptions = {
  browser: BrowserSvg,
  dao: DaoSvg,
  defi: DefiSvg,
  defiAlt: DefiAltSvg,
  eth: EthSvg,
  layers: LayersSvg,
  lock: LockSvg,
  login: LoginSvg,
  network: NetworkSvg,
  nft: NftSvg,
  noun: NounSvg,
  profile: ProfileSvg,
  system: SystemSvg
};
export function Visual({
  name,
  style
}) {
  const Component = svgOptions[name];
  return /*#__PURE__*/React.createElement(Component, {
    width: 60,
    height: 60,
    style: style
  });
}
