// @web3modal/ui-react-native/lib/commonjs/hooks/useAnimatedValue.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = require("react");
var _reactNative = require("react-native");
function useAnimatedValue(startValue, endValue, duration) {
  const valueRef = (0, _react.useRef)(new _reactNative.Animated.Value(0));
  const setStartValue = (0, _react.useCallback)(() => {
    _reactNative.Animated.timing(valueRef.current, {
      toValue: 0,
      useNativeDriver: false,
      duration: duration || 200
    }).start();
  }, [valueRef, duration]);
  const setEndValue = (0, _react.useCallback)(() => {
    _reactNative.Animated.timing(valueRef.current, {
      toValue: 1,
      useNativeDriver: false,
      duration: duration || 200
    }).start();
  }, [valueRef, duration]);
  const animatedValue = valueRef.current.interpolate({
    inputRange: [0, 1],
    outputRange: [startValue, endValue]
  });
  return {
    animatedValue,
    valueRef,
    setStartValue,
    setEndValue
  };
}
var _default = exports.default = useAnimatedValue;
