// @web3modal/ui-react-native/lib/commonjs/composites/wui-input-element/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.InputElement = InputElement;
var _reactNative = require("react-native");
var _wuiIcon = require("../../components/wui-icon");
var _useAnimatedValue = _interopRequireDefault(require("../../hooks/useAnimatedValue"));
var _useTheme = require("../../hooks/useTheme");
var _styles = _interopRequireDefault(require("./styles"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
const AnimatedPressable = _reactNative.Animated.createAnimatedComponent(_reactNative.Pressable);
function InputElement({
  icon,
  disabled,
  onPress,
  style
}) {
  const Theme = (0, _useTheme.useTheme)();
  const {
    animatedValue,
    setStartValue,
    setEndValue
  } = (0, _useAnimatedValue.default)(Theme['gray-glass-020'], Theme['gray-glass-030']);
  return /*#__PURE__*/React.createElement(AnimatedPressable, {
    disabled: disabled,
    onPress: onPress,
    onPressIn: setEndValue,
    onPressOut: setStartValue,
    hitSlop: 10,
    style: [_styles.default.container, {
      backgroundColor: disabled ? Theme['gray-glass-010'] : animatedValue
    }, style]
  }, /*#__PURE__*/React.createElement(_wuiIcon.Icon, {
    name: icon,
    size: "xxs",
    color: 'bg-200'
  }));
}
