// @web3modal/ethers-react-native/lib/commonjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "W3mAccountButton", {
  enumerable: true,
  get: function () {
    return _scaffoldReactNative.W3mAccountButton;
  }
});
Object.defineProperty(exports, "W3mButton", {
  enumerable: true,
  get: function () {
    return _scaffoldReactNative.W3mButton;
  }
});
Object.defineProperty(exports, "W3mConnectButton", {
  enumerable: true,
  get: function () {
    return _scaffoldReactNative.W3mConnectButton;
  }
});
Object.defineProperty(exports, "W3mNetworkButton", {
  enumerable: true,
  get: function () {
    return _scaffoldReactNative.W3mNetworkButton;
  }
});
Object.defineProperty(exports, "Web3Modal", {
  enumerable: true,
  get: function () {
    return _scaffoldReactNative.Web3Modal;
  }
});
exports.createWeb3Modal = createWeb3Modal;
Object.defineProperty(exports, "defaultConfig", {
  enumerable: true,
  get: function () {
    return _defaultConfig.defaultConfig;
  }
});
exports.useDisconnect = useDisconnect;
exports.useWeb3Modal = useWeb3Modal;
exports.useWeb3ModalAccount = useWeb3ModalAccount;
exports.useWeb3ModalError = useWeb3ModalError;
exports.useWeb3ModalEvents = useWeb3ModalEvents;
exports.useWeb3ModalProvider = useWeb3ModalProvider;
exports.useWeb3ModalState = useWeb3ModalState;
var _react = require("react");
var _valtio = require("valtio");
var _scaffoldUtilsReactNative = require("@web3modal/scaffold-utils-react-native");
var _scaffoldReactNative = require("@web3modal/scaffold-react-native");
var _defaultConfig = require("./utils/defaultConfig");
var _client = require("./client");
// -- Types -------------------------------------------------------------------

// -- Setup -------------------------------------------------------------------
let modal;
function createWeb3Modal(options) {
  if (!modal) {
    modal = new _client.Web3Modal({
      ...options,
      _sdkVersion: `react-native-ethers-${_scaffoldUtilsReactNative.ConstantsUtil.VERSION}`
    });
  }
  return modal;
}

// -- Hooks -------------------------------------------------------------------
function useWeb3Modal() {
  if (!modal) {
    throw new Error('Please call "createWeb3Modal" before using "useWeb3Modal" hook');
  }
  async function open(options) {
    await modal?.open(options);
  }
  async function close() {
    await modal?.close();
  }
  return {
    open,
    close
  };
}
function useWeb3ModalState() {
  if (!modal) {
    throw new Error('Please call "createWeb3Modal" before using "useWeb3ModalState" hook');
  }
  const [state, setState] = (0, _react.useState)(modal.getState());
  (0, _react.useEffect)(() => {
    const unsubscribe = modal?.subscribeState(newState => {
      if (newState) setState({
        ...newState
      });
    });
    return () => {
      unsubscribe?.();
    };
  }, []);
  return state;
}
function useWeb3ModalProvider() {
  const {
    provider,
    providerType
  } = (0, _valtio.useSnapshot)(_scaffoldUtilsReactNative.EthersStoreUtil.state);
  const walletProvider = provider;
  const walletProviderType = providerType;
  return {
    walletProvider,
    walletProviderType
  };
}
function useDisconnect() {
  async function disconnect() {
    await modal?.disconnect();
  }
  return {
    disconnect
  };
}
function useWeb3ModalAccount() {
  const {
    address,
    isConnected,
    chainId
  } = (0, _valtio.useSnapshot)(_scaffoldUtilsReactNative.EthersStoreUtil.state);
  return {
    address,
    isConnected,
    chainId
  };
}
function useWeb3ModalError() {
  const {
    error
  } = (0, _valtio.useSnapshot)(_scaffoldUtilsReactNative.EthersStoreUtil.state);
  return {
    error
  };
}
function useWeb3ModalEvents() {
  if (!modal) {
    throw new Error('Please call "createWeb3Modal" before using "useWeb3ModalEvents" hook');
  }
  const [event, setEvents] = (0, _react.useState)(modal.getEvent());
  (0, _react.useEffect)(() => {
    const unsubscribe = modal?.subscribeEvents(newEvent => {
      setEvents({
        ...newEvent
      });
    });
    return () => {
      unsubscribe?.();
    };
  }, []);
  return event;
}
