// @web3modal/core-react-native/lib/module/controllers/OptionsController.js
import { proxy, ref } from 'valtio';

// -- Types --------------------------------------------- //

// -- State --------------------------------------------- //
const state = proxy({
  projectId: '',
  sdkType: 'w3m',
  sdkVersion: 'react-native-undefined'
});

// -- Controller ---------------------------------------- //
export const OptionsController = {
  state,
  setClipboardClient(client) {
    state._clipboardClient = ref(client);
  },
  setProjectId(projectId) {
    state.projectId = projectId;
  },
  setIncludeWalletIds(includeWalletIds) {
    state.includeWalletIds = includeWalletIds;
  },
  setExcludeWalletIds(excludeWalletIds) {
    state.excludeWalletIds = excludeWalletIds;
  },
  setFeaturedWalletIds(featuredWalletIds) {
    state.featuredWalletIds = featuredWalletIds;
  },
  setTokens(tokens) {
    state.tokens = tokens;
  },
  setCustomWallets(customWallets) {
    state.customWallets = customWallets;
  },
  setEnableAnalytics(enableAnalytics) {
    state.enableAnalytics = enableAnalytics;
  },
  setSdkVersion(sdkVersion) {
    state.sdkVersion = sdkVersion;
  },
  isClipboardAvailable() {
    return !!state._clipboardClient;
  },
  copyToClipboard(value) {
    const client = state._clipboardClient;
    if (client) {
      client.setString(value);
    }
  }
};
