// @web3modal/ui-react-native/lib/commonjs/composites/wui-snackbar/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Snackbar = Snackbar;
var _reactNative = require("react-native");
var _wuiText = require("../../components/wui-text");
var _useTheme = require("../../hooks/useTheme");
var _wuiIconBox = require("../wui-icon-box");
var _styles = _interopRequireDefault(require("./styles"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function Snackbar({
  message,
  iconColor,
  icon,
  style
}) {
  const Theme = (0, _useTheme.useTheme)();
  return /*#__PURE__*/React.createElement(_reactNative.Animated.View, {
    style: [_styles.default.container, {
      backgroundColor: Theme['bg-175'],
      borderColor: Theme['gray-glass-005']
    }, style]
  }, /*#__PURE__*/React.createElement(_wuiIconBox.IconBox, {
    icon: icon,
    iconColor: iconColor,
    size: "sm",
    background: true
  }), /*#__PURE__*/React.createElement(_wuiText.Text, {
    variant: "paragraph-500",
    color: "fg-100",
    style: _styles.default.text
  }, message));
}
