// @web3modal/scaffold-react-native/lib/commonjs/modal/w3m-account-button/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.W3mAccountButton = W3mAccountButton;
var _valtio = require("valtio");
var _coreReactNative = require("@web3modal/core-react-native");
var _uiReactNative = require("@web3modal/ui-react-native");
function W3mAccountButton({
  balance,
  disabled,
  style,
  testID
}) {
  const {
    address,
    balance: balanceVal,
    balanceSymbol,
    profileImage,
    profileName
  } = (0, _valtio.useSnapshot)(_coreReactNative.AccountController.state);
  const {
    caipNetwork
  } = (0, _valtio.useSnapshot)(_coreReactNative.NetworkController.state);
  const networkImage = _coreReactNative.AssetUtil.getNetworkImage(caipNetwork);
  const showBalance = balance === 'show';
  return /*#__PURE__*/React.createElement(_uiReactNative.AccountButton, {
    onPress: _coreReactNative.ModalController.open,
    address: profileName ?? address,
    isProfileName: Boolean(profileName),
    networkSrc: networkImage,
    imageHeaders: _coreReactNative.ApiController._getApiHeaders(),
    avatarSrc: profileImage,
    disabled: disabled,
    style: style,
    balance: showBalance ? _coreReactNative.CoreHelperUtil.formatBalance(balanceVal, balanceSymbol) : '',
    testID: testID
  });
}
