// @web3modal/ui-react-native/lib/module/layout/wui-flex/index.js
import { View } from 'react-native';
import { UiUtil } from '../../utils/UiUtil';
export function FlexView(props) {
  const styles = {
    flexDirection: props.flexDirection,
    flexWrap: props.flexWrap,
    flexGrow: props.flexGrow,
    flexShrink: props.flexShrink,
    alignItems: props.alignItems,
    alignSelf: props.alignSelf,
    justifyContent: props.justifyContent,
    paddingTop: props.padding && UiUtil.getSpacingStyles(props.padding, 0),
    paddingRight: props.padding && UiUtil.getSpacingStyles(props.padding, 1),
    paddingBottom: props.padding && UiUtil.getSpacingStyles(props.padding, 2),
    paddingLeft: props.padding && UiUtil.getSpacingStyles(props.padding, 3),
    marginTop: props.margin && UiUtil.getSpacingStyles(props.margin, 0),
    marginRight: props.margin && UiUtil.getSpacingStyles(props.margin, 1),
    marginBottom: props.margin && UiUtil.getSpacingStyles(props.margin, 2),
    marginLeft: props.margin && UiUtil.getSpacingStyles(props.margin, 3)
  };
  return /*#__PURE__*/React.createElement(View, {
    style: [styles, props.style]
  }, props.children);
}
