// @web3modal/ui-react-native/lib/module/composites/wui-qr-code/index.js
import { useMemo } from 'react';
import { View } from 'react-native';
import Svg from 'react-native-svg';
import { Icon } from '../../components/wui-icon';
import { Image } from '../../components/wui-image';
import { Shimmer } from '../../components/wui-shimmer';
import { QRCodeUtil } from '../../utils/QRCodeUtil';
import { BorderRadius, LightTheme, Spacing } from '../../utils/ThemeUtil';
import styles from './styles';
export function QrCode({
  size,
  uri,
  imageSrc,
  testID
}) {
  const Theme = LightTheme;
  const containerPadding = Spacing.l;
  const qrSize = size - containerPadding * 2;
  const dots = useMemo(() => uri ? QRCodeUtil.generate(uri, qrSize, qrSize / 4) : [], [uri, qrSize]);
  const shimmerTemplate = () => {
    return /*#__PURE__*/React.createElement(Shimmer, {
      width: size,
      height: size,
      borderRadius: BorderRadius.l
    });
  };
  const logoTemplate = () => {
    if (imageSrc) {
      return /*#__PURE__*/React.createElement(Image, {
        source: imageSrc,
        style: [styles.icon, {
          height: qrSize / 4,
          width: qrSize / 4,
          borderRadius: qrSize / 16
        }]
      });
    }
    return /*#__PURE__*/React.createElement(Icon, {
      name: "walletConnect",
      color: "accent-100",
      height: qrSize / 3.5,
      width: qrSize / 3.5,
      style: styles.icon
    });
  };
  return uri ? /*#__PURE__*/React.createElement(View, {
    style: [styles.container, {
      height: size,
      width: size,
      backgroundColor: Theme['bg-100'],
      padding: containerPadding
    }],
    testID: testID
  }, /*#__PURE__*/React.createElement(Svg, {
    height: qrSize,
    width: qrSize
  }, dots), logoTemplate()) : shimmerTemplate();
}
